<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Dissecting UMAP visualizations | Sam Morabito </title> <meta name="author" content="Sam Morabito"> <meta name="description" content="Computing a bunch of UMAPs using different parameters."> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://smorabit.github.io/blog/2020/umap/"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?bf50d6d9dd867d3e0f3b0add94449649"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sam </span> Morabito </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fa-solid fa-moon"></i> <i class="fa-solid fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Dissecting UMAP visualizations</h1> <p class="post-meta"> August 09, 2020 </p> <p class="post-tags"> <a href="/blog/2020"> <i class="fa-solid fa-calendar fa-sm"></i> 2020 </a>   ·   <a href="/blog/tag/singlecell"> <i class="fa-solid fa-hashtag fa-sm"></i> singlecell</a>   <a href="/blog/tag/rna"> <i class="fa-solid fa-hashtag fa-sm"></i> rna</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>In this blog post I aim to showcase the Uniform Manifold Approximation and Projection (UMAP) algorithm from a practical standpoint by experimenting with its various hyperparameters in a single cell RNA-seq (scRNA-seq) dataset of <code class="language-plaintext highlighter-rouge">3,000</code> cells uniformly sampled from a mouse brain dataset. UMAP is an <a href="https://en.wikipedia.org/wiki/Unsupervised_learning" rel="external nofollow noopener" target="_blank">unsupervised learning</a> algorithm for constructing low-dimensional manifolds in high-dimensional datasets, and is commonly applied for the purpose of data visualization. UMAP and other dimensionality reduction techniques help with interpretability of complex datasets. For example, in the case of scRNA-seq analysis, UMAP helps us understand the differences and similarities between different cells by reducing their transcriptomes from ~20k+ dimensions down to two dimensions that we can inspect visually. Scanpy offets a nice <a href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html" rel="external nofollow noopener" target="_blank">tutorial</a> for getting started with scRNA-seq analysis, including a dataset of peripheral mononuclear blood cells (PBMCs).</p> <p>In their 2018 <a href="https://arxiv.org/abs/1802.03426" rel="external nofollow noopener" target="_blank">arXiv paper</a>, McInnes &amp; Healy introduce UMAP and argue that it is competitive with t-distributed Stochastic Neighbor Embedding <a href="https://en.wikipedia.org/wiki/T-distributed_stochastic_neighbor_embedding" rel="external nofollow noopener" target="_blank">(t-SNE)</a> in terms of visualization quality while offering superior run time performance. Additionally, the authors argue that the global structure of the data is better preserved in UMAP compared to t-SNE.</p> <p>While UMAP can be used for general-purpose dimensionality reduction, in single-cell genomics field it is usually applied to data that has already been reduced using a linear transformation such as <a href="https://en.wikipedia.org/wiki/Principal_component_analysis" rel="external nofollow noopener" target="_blank">principal component analysis (PCA)</a>. UMAP has been integrated in almost every single-cell data analysis toolkit, including <a href="https://satijalab.org/seurat/" rel="external nofollow noopener" target="_blank">Seurat</a> and <a href="https://scanpy.readthedocs.io/en/stable/" rel="external nofollow noopener" target="_blank">Scanpy</a>.</p> <h3 id="what-does-a-umap-plot-look-like">What does a UMAP plot look like?</h3> <p>The following scatter plot shows the dataset of <code class="language-plaintext highlighter-rouge">3,000</code> cells and <code class="language-plaintext highlighter-rouge">19,998</code> genes that has been reduced to <code class="language-plaintext highlighter-rouge">3,000</code> cells (dots) and <code class="language-plaintext highlighter-rouge">2</code> UMAP dimensions, visualized in the plot below. Each cell is colored by cluster assignment from <a href="https://www.nature.com/articles/s41598-019-41695-z" rel="external nofollow noopener" target="_blank">Leiden clustering</a> on the PCA reduced dataset.</p> <div class="img"> <img src="/assets/img/umap/umap_default.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <div class="caption"> UMAP with default parameters colored with Leiden clusters </div> <p>In practice, computing a UMAP with Scanpy is very easy, and you don’t necessarily need to think about hyperparameters at all. The following line of code constructed the above UMAP on the scRNA-seq dataset.</p> <figure class="highlight"><pre><code class="language-python" data-lang="python"><span class="n">sc</span><span class="p">.</span><span class="n">tl</span><span class="p">.</span><span class="nf">umap</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span></code></pre></figure> <p>In many cases it is okay to use UMAP’s default parameters, but if you are like me you may be curious what these default settings are and how they are influencing your interpretation of the data. For single-cell analysis, cells should group together in UMAP space based on cell type identity. Cell types must be carefully annotated by inspecting the distribution of gene expression values in literature curated lists of canonical cell type marker genes, and by using statistical tests to identify up- and down-regulated genes in each cell population. UMAP visualizations of these genes can be helpful to diagnose problematic data processing or erroneous cell type annotations.</p> <h3 id="playing-with-umap-hyperparameters">Playing with UMAP hyperparameters</h3> <p>Here I show the results of testing different values for seven different UMAP hyperparameters. The following hyperparameter descriptions are taken from the <a href="https://scanpy.readthedocs.io/en/stable/api/scanpy.tl.umap.html" rel="external nofollow noopener" target="_blank">Scanpy documentation</a>.</p> <ul> <li> <code class="language-plaintext highlighter-rouge">n_neighbors</code>: A value between 2 and 100, representing the number of neighboring data points used for manifld approximation. Larger values give a manifold with a more global view of the dataset, while smaller values preserve more of the local structures.</li> <li> <code class="language-plaintext highlighter-rouge">min_dist</code>: The minimum distance between two points in the UMAP embedding.</li> <li> <code class="language-plaintext highlighter-rouge">spread</code>: A scaling factor for distance between embedded points.</li> <li> <code class="language-plaintext highlighter-rouge">gamma</code>: Weighting applied to negative samples in low dimensional embedding optimization.</li> <li> <code class="language-plaintext highlighter-rouge">alpha</code>: The initial learning rate for UMAP optimization.</li> <li> <code class="language-plaintext highlighter-rouge">maxiter</code>: The number of iterations for UMAP optimization.</li> <li> <code class="language-plaintext highlighter-rouge">negative_sample_rate</code>: The number of negative edge/1-simplex samples to use per positive edge/1-simplex sample in optimizing the low dimensional embedding.</li> </ul> <h3 id="test-1-n_neighbors">Test 1: <code class="language-plaintext highlighter-rouge">n_neighbors</code> </h3> <div class="img"> <img src="/assets/img/umap/umap_neighbors_illustrator.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <div class="col three caption"> Parameter sweep for min_dist and spread </div> <p>The resulting UMAPs look quite similar, but they do have some clear distinctions. As n_neighbors increases, the clusters are encroaching upon each other. Interestingly, the blue cluster on the left side of the UMAPs appears to be completely separated from the rest of the clusters only in the left-most UMAP. Larger values of min_dist may cause rare cell populations to get mixed into other cell types, since their spatial assignment would be more influenced by the global data landscape.</p> <h3 id="test-2-min_dist-and-spread">Test 2: <code class="language-plaintext highlighter-rouge">min_dist</code> and <code class="language-plaintext highlighter-rouge">spread</code> </h3> <div class="img"> <img src="/assets/img/umap/umap_spread_dist_edit.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <div class="col three caption"> Parameter sweep for min_dist and spread </div> <p>Here we can see that in general <code class="language-plaintext highlighter-rouge">min_dist</code> and <code class="language-plaintext highlighter-rouge">spread</code> influence the distance between points in the UMAP embedding, and that large values of <code class="language-plaintext highlighter-rouge">min_dist</code> results in the UMAP looking like a messy blob. In contrast, larger values of <code class="language-plaintext highlighter-rouge">spread</code> yield tighter clustering. In this dataset lower values of <code class="language-plaintext highlighter-rouge">min_dist</code> seem desirable in order to separate different cell populations in low dimensional space.</p> <h3 id="test-3-gamma-and-alpha">Test 3: <code class="language-plaintext highlighter-rouge">gamma</code> and <code class="language-plaintext highlighter-rouge">alpha</code> </h3> <div class="img"> <img src="/assets/img/umap/umap_alpha_gamma_edit.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <div class="col three caption"> Parameter sweep for gamma and alpha </div> <p><code class="language-plaintext highlighter-rouge">alpha</code> and <code class="language-plaintext highlighter-rouge">gamma</code> do not appear to have very much influence on the resulting UMAPs. Perhaps more extreme values need to be tested, or perhaps a larger dataset should be used.</p> <h3 id="test-4-maxiter-and-negative_sample_rate">Test 4: <code class="language-plaintext highlighter-rouge">maxiter</code> and <code class="language-plaintext highlighter-rouge">negative_sample_rate</code> </h3> <div class="img"> <img src="/assets/img/umap/umap_sample_rate_edit.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <div class="col three caption"> Parameter sweep negative sample rate and number of training epochs </div> <p>Similar to Test 3, <code class="language-plaintext highlighter-rouge">maxiter</code> and <code class="language-plaintext highlighter-rouge">negative_sample_rate</code> do not appear to have very much influence on the resulting UMAPs. This is helpful to know that it does not take a large amount of training iterations to achieve a decent UMAP.</p> <h3 id="conclusion">Conclusion</h3> <p>In most cases it seems like it is okay to carry on using the default UMAP hyperparameters. If anything should be changed to best suit a certain dataset, the parameters that are going to have the most influence on the output are n_neighbors, and spread/min_dist. Perhaps it would be more informative to test a wider range of values, and to use different sized datasets (future blog post maybe). Take note that different software packages may have slightly different default parameters for UMAP.</p> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Sam Morabito. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?2930004b8d7fcd0a8e00fdcfc8fc9f24"></script> <script defer src="/assets/js/common.js?4a129fbf39254905f505c7246e641eaf"></script> <script defer src="/assets/js/copy_code.js?7254ae07fe9cc5f3a10843e1c0817c9c" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>