<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Single-cell RNA-seq essentials (legacy) | Sam Morabito </title> <meta name="author" content="Sam Morabito"> <meta name="description" content="Tutorial coviering the basics of scRNA-seq clustering analysis with Seurat."> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://smorabit.github.io/blog/2021/single-cell-essentials/"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?bf50d6d9dd867d3e0f3b0add94449649"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sam </span> Morabito </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fa-solid fa-moon"></i> <i class="fa-solid fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Single-cell RNA-seq essentials (legacy)</h1> <p class="post-meta"> May 23, 2021 </p> <p class="post-tags"> <a href="/blog/2021"> <i class="fa-solid fa-calendar fa-sm"></i> 2021 </a>   ·   <a href="/blog/tag/singlecell"> <i class="fa-solid fa-hashtag fa-sm"></i> singlecell</a>   <a href="/blog/tag/rna"> <i class="fa-solid fa-hashtag fa-sm"></i> rna</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Note: this is a LEGACY tutorial, meaning that it was written several years ago and is using potentially outdated software.</p> <h1 id="introduction">Introduction</h1> <p>This document is the first in a series of tutorials covering the essentials of single-cell transcriptomics analysis. The following figure from Luecken &amp; Theis illustrates many of the analysis techniques that are commonly employed in my lab and by many others for single-cell transcriptomics analysis.</p> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/sc-overview.png"> </div> </div> <div class="caption"> Overview of single-cell RNA-seq analysis in Luecken &amp; Theis 2019 </div> <p>The following topics are covered in this tutorial:</p> <ul> <li>Load UMI counts matrix into R/Seurat.</li> <li>Quality Control</li> <li>Normalization</li> <li>Feature selection</li> <li>Dimensionality Reduction</li> <li>Clustering</li> <li>Visualization</li> </ul> <h2 id="seurat">Seurat</h2> <p>The following section covers the absolute basics of single-cell analysis using the R package <a href="https://satijalab.org/seurat/" rel="external nofollow noopener" target="_blank">Seurat</a>. Most of this tutorial is inspired by <a href="https://satijalab.org/seurat/v3.2/seurat_obj3k_tutorial.html" rel="external nofollow noopener" target="_blank">Seurat’s clustering tutorial</a>, however we will be using a dataset from the human brain since that is much more relevant to the lab’s research.</p> <h3 id="loading-data-into-seurat">Loading data into Seurat</h3> <p>In many cases, we work with single-cell data generated from the 10X Genomics platform. Single-cell analysis packages such as Seurat and Scanpy make it easy to load UMI counts matrices from the output of cellranger. The following commands create a Seurat object from the output of cellranger:</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span><span class="w">

</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Read10X</span><span class="p">(</span><span class="n">data.dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cellranger_aggr_dir/outs/filtered_feature_bc_matrix/'</span><span class="p">)</span><span class="w">

</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CreateSeuratObject</span><span class="p">(</span><span class="w">
  </span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="p">,</span><span class="w">
  </span><span class="n">project</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'tutorial'</span><span class="p">,</span><span class="w">
  </span><span class="n">min.cells</span><span class="w"> </span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w">
  </span><span class="n">min.features</span><span class="o">=</span><span class="m">200</span><span class="w">
</span><span class="p">)</span></code></pre></figure> </details> <p>For this tutorial I am using a published dataset from a study of TREM2 in AD using single-cell transcriptomics in human and mouse samples <a href="https://pubmed.ncbi.nlm.nih.gov/31932797/" rel="external nofollow noopener" target="_blank">(link to study)</a>. In this case a UMI counts matrix is stored separately for each sample in <code class="language-plaintext highlighter-rouge">.mtx</code> format, so we have to load each of these files before merging into one big Seurat object. This is a more general use case than using the <code class="language-plaintext highlighter-rouge">Read10X</code> function.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">Matrix</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span><span class="w">
</span><span class="n">theme_set</span><span class="p">(</span><span class="n">theme_cowplot</span><span class="p">())</span><span class="w">

</span><span class="n">data_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'trem2_nmed/'</span><span class="w">
</span><span class="n">files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dir</span><span class="p">(</span><span class="n">data_dir</span><span class="p">)</span><span class="w">
</span><span class="n">files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">files</span><span class="p">[</span><span class="n">grepl</span><span class="p">(</span><span class="s1">'matrix'</span><span class="p">,</span><span class="w"> </span><span class="n">files</span><span class="p">)]</span><span class="w">
</span><span class="n">file_stems</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">str_replace_all</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="w"> </span><span class="s1">'_matrix.mtx'</span><span class="p">,</span><span class="w"> </span><span class="s1">''</span><span class="p">)</span><span class="w">

</span><span class="c1"># construct a list of seurat objects for each sample by iteratively loading each file</span><span class="w">
</span><span class="n">seurat_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">file_stems</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">file</span><span class="p">){</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w">
  </span><span class="n">X</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readMM</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="n">file</span><span class="p">,</span><span class="s1">'_matrix.mtx'</span><span class="p">))</span><span class="w">

  </span><span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="n">file</span><span class="p">,</span><span class="s1">'_features.tsv'</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
  </span><span class="n">barcodes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="n">file</span><span class="p">,</span><span class="s1">'_barcodes.tsv'</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="n">colnames</span><span class="p">(</span><span class="n">X</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">barcodes</span><span class="o">$</span><span class="n">V1</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">X</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genes</span><span class="o">$</span><span class="n">V2</span><span class="w">

  </span><span class="n">cur_seurat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CreateSeuratObject</span><span class="p">(</span><span class="w">
    </span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">X</span><span class="p">,</span><span class="w">
    </span><span class="n">project</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"tutorial"</span><span class="w">
  </span><span class="p">)</span><span class="w">
  </span><span class="n">cur_seurat</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">SampleID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file</span><span class="w">
  </span><span class="n">cur_seurat</span><span class="w">
</span><span class="p">})</span><span class="w">

</span><span class="c1"># merge into one big seurat object</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">seurat_list</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">seurat_list</span><span class="p">[</span><span class="m">2</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seurat_list</span><span class="p">)])</span><span class="w">

</span><span class="c1"># add metadata</span><span class="w">
</span><span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s1">'meta.csv'</span><span class="p">))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">meta</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="o">$</span><span class="n">Sample.ID.in.snRNA.seq</span><span class="w">

</span><span class="n">meta.data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="p">[</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">SampleID</span><span class="p">,]</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">m</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">meta.data</span><span class="p">)){</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="o">@</span><span class="n">meta.data</span><span class="p">[[</span><span class="n">m</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta.data</span><span class="p">[[</span><span class="n">m</span><span class="p">]]</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># remove individual seurat objects to save memory</span><span class="w">
</span><span class="n">rm</span><span class="p">(</span><span class="n">seurat_list</span><span class="p">);</span><span class="w"> </span><span class="n">gc</span><span class="p">();</span></code></pre></figure> </details> <p>Let’s get a sense for the size of our dataset by plotting the number of cells in each sample.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># use table function to get the number of cells in each Sample as a dataframe</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">SampleID</span><span class="p">)))</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'SampleID'</span><span class="p">,</span><span class="w"> </span><span class="s1">'n_cells'</span><span class="p">)</span><span class="w">

</span><span class="c1"># bar plot of the number of cells in each sample</span><span class="w">
</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">n_cells</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">reorder</span><span class="p">(</span><span class="n">SampleID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">n_cells</span><span class="p">),</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">SampleID</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">NoLegend</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">RotatedAxis</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylab</span><span class="p">(</span><span class="nf">expression</span><span class="p">(</span><span class="n">italic</span><span class="p">(</span><span class="n">N</span><span class="p">)[</span><span class="n">cells</span><span class="p">]))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlab</span><span class="p">(</span><span class="s1">'Sample ID'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s1">'Total cells:'</span><span class="p">,</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">n_cells</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="w">
    </span><span class="n">panel.grid.minor</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">panel.grid.major.y</span><span class="o">=</span><span class="n">element_line</span><span class="p">(</span><span class="n">colour</span><span class="o">=</span><span class="s2">"lightgray"</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
  </span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_cells_per_sample.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_cells_per_sample.png"> </div> </div> <div class="caption"> Number of cells per sample before filtering </div> <p>We can see that the number of cells varies quite a bit between samples, the largest number of cells one sample being <code class="language-plaintext highlighter-rouge">9718</code> and the smallest being <code class="language-plaintext highlighter-rouge">1075</code>. Based on how the experiment was set up, we reasonably expect <code class="language-plaintext highlighter-rouge">10,000</code> cells as the maximum for a single sample, and a minimum of a few thousand, so overall this is not out of the ordinary. Next we will remove some outliers that do not pass our quality control criteria.</p> <h3 id="quality-control">Quality Control</h3> <p>In this section we will remove low quality cells based on several quality control criteria, such as the percentage of reads in mitochondrial genes, the number of genes detected per cell, and the number of UMIs detected per cell. First we compute the percentage of mitochondrial reads for each cell, and then we plot the distribution of these metrics in each sample.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># calculate the percentage of mitochondrial reads per cell</span><span class="w">
</span><span class="n">seurat_obj</span><span class="p">[[</span><span class="s2">"percent.mt"</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">PercentageFeatureSet</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"^MT-"</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot distributions of QC metrics, grouped by SampleID</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_qc.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">VlnPlot</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">features</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"nFeature_RNA"</span><span class="p">,</span><span class="w"> </span><span class="s2">"nCount_RNA"</span><span class="p">,</span><span class="w"> </span><span class="s2">"percent.mt"</span><span class="p">),</span><span class="w">
  </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'SampleID'</span><span class="p">,</span><span class="w">
  </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">pt.size</span><span class="o">=</span><span class="m">0</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_qc.png"> </div> </div> <div class="caption"> Distributions of QC metrics in each sample </div> <p>These distributions provide insight into the overall quality of these samples. For example, we observe that sample <strong>AD8</strong> has much lower UMI and gene detection than other samples, as well as a much higher percent of mitochondrial reads. Considering the relative low quality of sample <strong>AD8</strong>, it might be indicative that something went wrong with the single-cell RNA-seq experiment itself, and that it should be removed from the analysis entirely. In this published dataset, the authors included all of the cells before quality control, so we have to apply the QC filters ourselves.</p> <p>Here we apply QC filtering such that the downstream analysis only contains only very high quality cells. I want to emphasise how important this step is, as outliers can highly influence all of the downstream analysis and interpretation. Later on if something doesn’t look right, this is always a good step to revisit.</p> <p>Additionally, this step serves as a great example of how often data analysis is both an art and a science. There isn’t a one-size fits all solution to properly filtering every single-cell dataset, meaning that some level of personal judgement is required.</p> <p>Here I will apply the following filters:</p> <ul> <li>Remove cells with more than 15% mitochondrial reads.</li> <li>Remove cells with greater than <code class="language-plaintext highlighter-rouge">30,000</code> UMIs and fewer than <code class="language-plaintext highlighter-rouge">250</code> UMIs.</li> </ul> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># apply filter</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">subset</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">nCount_RNA</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">250</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">nCount_RNA</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">30000</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">percent.mt</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot the number of cells in each sample post filtering</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">SampleID</span><span class="p">)))</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'SampleID'</span><span class="p">,</span><span class="w"> </span><span class="s1">'n_cells'</span><span class="p">)</span><span class="w">
</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">n_cells</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">reorder</span><span class="p">(</span><span class="n">SampleID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">n_cells</span><span class="p">),</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">SampleID</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">NoLegend</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">RotatedAxis</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylab</span><span class="p">(</span><span class="nf">expression</span><span class="p">(</span><span class="n">italic</span><span class="p">(</span><span class="n">N</span><span class="p">)[</span><span class="n">cells</span><span class="p">]))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlab</span><span class="p">(</span><span class="s1">'Sample ID'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s1">'Total cells post-filtering:'</span><span class="p">,</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">n_cells</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="w">
    </span><span class="n">panel.grid.minor</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">panel.grid.major.y</span><span class="o">=</span><span class="n">element_line</span><span class="p">(</span><span class="n">colour</span><span class="o">=</span><span class="s2">"lightgray"</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
  </span><span class="p">)</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_cells_per_sample_filtered.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_cells_per_sample_filtered.png"> </div> </div> <div class="caption"> Number of cells per sample post-filtering </div> <p>Using these filters we removed <code class="language-plaintext highlighter-rouge">20,125</code> low-quality cells, retaining a total of <code class="language-plaintext highlighter-rouge">94,847</code> cells for downstream analysis. It looks like the paper actually has a more stringent QC cutoff, retaining a total of <code class="language-plaintext highlighter-rouge">66,311</code> nuclei.</p> <h3 id="normalization">Normalization</h3> <p>In this section we apply a logarithmic transformation and apply a scaling factor to the UMI counts matrix. We then scale the data to have a mean expression of 0 for each feature, and a variance of 1 (this is standard for many machine learning pre-processing pipelines).</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># log normalize data</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">NormalizeData</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">normalization.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"LogNormalize"</span><span class="p">,</span><span class="w">
  </span><span class="n">scale.factor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10000</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># scale data:</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ScaleData</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">features</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">))</span></code></pre></figure> </details> <h3 id="feature-selection">Feature Selection</h3> <p>In this section we identify a set of genes that are highly variable across the dataset. These genes tend to be more cell type and cell population specific, so we expect that they have high expression levels in some cells and low in others. We will use these genes for downstream analysis such as dimensionality reduction and clustering. Much like the previous QC step, the number of features to select is highly subject to each dataset.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FindVariableFeatures</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">selection.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"vst"</span><span class="p">,</span><span class="w">
  </span><span class="n">nfeatures</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4000</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">LabelPoints</span><span class="p">(</span><span class="w">
  </span><span class="n">VariableFeaturePlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">),</span><span class="w">
  </span><span class="n">points</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">VariableFeatures</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">),</span><span class="m">10</span><span class="p">),</span><span class="w">
  </span><span class="n">repel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="w">
</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"bottom"</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_variable_features.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_variable_features.png"> </div> </div> <div class="caption"> Selection of highly variable genes </div> <h3 id="linear-dimensionality-reduction">Linear Dimensionality Reduction</h3> <p>In this section we perform Principal Components Analysis (PCA) to construct a linear dimensionality reduction of the dataset. Specifically, we perform PCA on the scaled dataset using only the genes that we have identified as highly variable in the previous section.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">RunPCA</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">features</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">VariableFeatures</span><span class="p">(</span><span class="n">object</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seurat_obj</span><span class="p">),</span><span class="w">
  </span><span class="n">npcs</span><span class="o">=</span><span class="m">50</span><span class="w">
</span><span class="p">)</span></code></pre></figure> </details> <p>Now that we have constructed a PCA matrix, we can inspect the genes that contribute the most to the variance of the PCs.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># plot the top genes contributing to the first 3 PCs</span><span class="w">
</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">VizDimLoadings</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"pca"</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">3</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_pca_loadings.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_pca_loadings.png"> </div> </div> <div class="caption"> Top genes in PCs 1-3 </div> <p>We can also visualize the data as a scatter plot showing each cell in PCA space.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># PCA scatter plot colored by SampleID</span><span class="w">
</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DimPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"pca"</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'SampleID'</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_pca_scatter.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_pca_scatter.png"> </div> </div> <div class="caption"> Scatter plot of PC1 vs PC2 colored by sample ID </div> <p>This scatter plot shows all <code class="language-plaintext highlighter-rouge">~90k</code> cells in PCA space. While there are some patterns that we can observe here, the cells do not form easily distinguishable clusters. Applying a non-linear dimensionality reduction on top of this PCA transformation should yield better results for visualization and clustering purposes.</p> <p>For downstream analysis, we do not necessarily need the entire PCA matrix. We can use a number of methods to identify the PCs that contain most of the complexity of the dataset, and discard the remaining PCs. First we can simply visualize heatmaps of the PCA matrix.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># plot heatmaps for first 16 PCs</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_pca_heatmap.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">DimHeatmap</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">16</span><span class="p">,</span><span class="w"> </span><span class="n">cells</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">500</span><span class="p">,</span><span class="w"> </span><span class="n">balanced</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_pca_heatmap.png"> </div> </div> <div class="caption"> PCA heatmaps </div> <p>Now we inspect the standard deviation of each PC in a ‘elbow plot’.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_pca_elbow.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">ElbowPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">ndims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_pca_elbow.png"> </div> </div> <div class="caption"> PCA elbow plot </div> <p>Based on these plots we can decide how many PCs to retain for downstream analysis. Once again, there is not a one size fits all solution to determining the number of PCs for a particular single-cell dataset, so you may have to try using different numbers of PCs in the downstream analysis to figure out what is appropriate. Based on the elbow plot, I am going to use a cutoff of 30 PCs for this dataset.</p> <h3 id="clustering-and-non-linear-dimensionality-reduction">Clustering and non-linear dimensionality reduction</h3> <p>Here we cluster the cells using a graph-based approach within PCA space, and then apply non-linear dimensionality reductions for further visualization purposes. An important parameter for clustering is <code class="language-plaintext highlighter-rouge">resolution</code>, where a higher value of this parameter yields a larger number of clusters. For the purpose of this tutorial I am using <code class="language-plaintext highlighter-rouge">resolution=0.5</code>.</p> <p>Following clustering we apply <a href="https://arxiv.org/abs/1802.03426" rel="external nofollow noopener" target="_blank">UMAP</a> and <a href="https://en.wikipedia.org/wiki/T-distributed_stochastic_neighbor_embedding" rel="external nofollow noopener" target="_blank">t-SNE</a> for non-linear dimensionality reductions. These algorithms are pretty complex, so you may be interested in reading up about them, however it is not necessary to know how they work in detail to use them effectively. The overall goal of these approaches are to construct low-dimensional manifolds of high-dimensional datasets such that data entries (single cells in this case) that are similar are closer together in manifold space. Of course, there are many more manifold learning approaches aside from UMAP and t-SNE, each with their own pros and cons. UMAP is a newer approach compared to t-SNE, and it generally does a better job at preserving the global structure of the data instead of just local structures. Both t-SNE and UMAP have a variety of hyperparameters that can be tweaked ad infinitum, but for this tutorial (and in most use cases) we can use the default parameters. I actually have a short <a href="https://smorabit.github.io/blog/2020/umap/">blog post</a> going over UMAP hyperparameters, and for the most part the default parameters are suitable.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># KNN and clustering</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FindNeighbors</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FindClusters</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">resolution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w">

</span><span class="c1"># non-linear reductions (UMAP &amp; t-SNE)</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">RunUMAP</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">RunTSNE</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span></code></pre></figure> </details> <p>Now let’s look at our clusters using our UMAP and t-SNE embeddings.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">umap_theme</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">theme</span><span class="p">(</span><span class="w">
  </span><span class="n">axis.line</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">axis.text.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">axis.text.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">axis.title.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">axis.title.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">panel.background</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">panel.border</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">panel.grid.major</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
  </span><span class="n">panel.grid.minor</span><span class="o">=</span><span class="n">element_blank</span><span class="p">()</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_umap_clusters.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">DimPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"umap"</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'seurat_clusters'</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">umap_theme</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">NoLegend</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'UMAP colored by seurat clusters'</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_tsne_clusters.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">DimPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"tsne"</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'seurat_clusters'</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
 </span><span class="n">umap_theme</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">NoLegend</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'t-SNE colored by seurat clusters'</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_tsne_clusters.png"> </div> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_umap_clusters.png"> </div> </div> <div class="caption"> Left: t-SNE, Right: UMAP </div> <p>By coloring these plots by their cluster assignment, we can immediately see that both methods do a decent job at spatially separating cells by their clusters in this low-dimensional space. In general we don’t expect a perfect separation of the clusters in this space. Qualitatively, the UMAP plot separates the clusters further apart from one another, while the t-SNE plot looks more like a bunch of blobs stuck together. In this particular t-SNE I can tell that there is some funky things going on, for example part of cluster 2 is inside of cluster 0.</p> <h3 id="cell-type-annotation">Cell-type annotation</h3> <p>Up to this point, we have only needed to use our data scientist hat to understand the different parts of the single-cell data analysis workflow. We have gone from a tangled mess of billions of nucleotide sequences to a much more organized and interpretable format consisting of a cells by genes expression matrix, a linear dimensionality reduction matrix, clusters, and a two-dimensional data manifold. So now it is our job to start interpreting the biology within this datase</p> <p>We have to consider some of the known biology from published literature with regards to the samples that were sequenced for this analysis. Zhou et al. sequenced postmortem human brain samples from the dorsolateral prefrontal cortex (DLPFC). Based on what we know about the cellular composition of the DLPFC, we expect to primarily recover neurons and glia. Since we have transcriptomic data individual cells, we can actually provide much more precise annotations than neuronal/glia. You may wish to consult a neurobiology textbook to get a better idfea of what cell types you expect to find in the DLPFC and in other regions of the brain.</p> <p>In this section we use a literature-curated list of known cell-type marker genes to provide a cell type annotation to each cluster. A critical assumption of this method is that all cells within the same cluster are of the same cell type. This assumption mostly holds true with a high enough resolution, but of course there are some exceptions that we have to live with. Here we visualize the gene expression of these marker genes directly on the UMAP embedding, as we visualize the distributions of these marker genes in each cluster.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># set up list of canonical cell type markers</span><span class="w">
</span><span class="n">canonical_markers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
  </span><span class="s1">'Astrocyte'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'GFAP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'AQP4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SLC1A2'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Pan-neuronal'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'SNAP25'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SYT1'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Excitatory Neuron'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'SLC17A7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SATB2'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Inhibitory Neuron'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'GAD1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GAD2'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Microglia'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'CSF1R'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CD74'</span><span class="p">,</span><span class="w"> </span><span class="s1">'P2RY12'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Oligodendrocyte'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'MOBP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MBP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MOG'</span><span class="p">),</span><span class="w">
  </span><span class="s1">'Olig. Progenitor'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'PDGFRA'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CSPG4'</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot heatmap:</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridis</span><span class="p">)</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_canonical_marker_heatmap.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">)</span><span class="w">
</span><span class="n">DoHeatmap</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="w"> </span><span class="o">=</span><span class="s2">"seurat_clusters"</span><span class="p">,</span><span class="w"> </span><span class="n">features</span><span class="o">=</span><span class="nf">as.character</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">canonical_markers</span><span class="p">)))</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># create feature plots, cutoff expression values for the 98th and 99th percentile</span><span class="w">
</span><span class="n">plot_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FeaturePlot</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">features</span><span class="o">=</span><span class="n">unlist</span><span class="p">(</span><span class="n">canonical_markers</span><span class="p">),</span><span class="w">
  </span><span class="n">combine</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">cols</span><span class="o">=</span><span class="n">viridis</span><span class="p">(</span><span class="m">256</span><span class="p">),</span><span class="w">
  </span><span class="n">max.cutoff</span><span class="o">=</span><span class="s1">'q98'</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># apply theme to each feature plot</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">plot_list</span><span class="p">)){</span><span class="w">
  </span><span class="n">plot_list</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_list</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">umap_theme</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">NoLegend</span><span class="p">()</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_canonical_marker_featurePlot.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">)</span><span class="w">
</span><span class="n">CombinePlots</span><span class="p">(</span><span class="n">plot_list</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_canonical_marker_featurePlot.png"> </div> </div> <div class="caption"> Feature plots of canonical marker genes. Yellow = high expression, Blue = low expression. </div> <p>Next we can plot the distributions of these genes in each cluster.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="k">for</span><span class="p">(</span><span class="n">celltype</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">canonical_markers</span><span class="p">)){</span><span class="w">

  </span><span class="n">print</span><span class="p">(</span><span class="n">celltype</span><span class="p">)</span><span class="w">
  </span><span class="n">cur_features</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">canonical_markers</span><span class="p">[[</span><span class="n">celltype</span><span class="p">]]</span><span class="w">

  </span><span class="c1"># plot distributions for marker genes:</span><span class="w">
  </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">VlnPlot</span><span class="p">(</span><span class="w">
    </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
    </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'seurat_clusters'</span><span class="p">,</span><span class="w">
    </span><span class="n">features</span><span class="o">=</span><span class="n">cur_features</span><span class="p">,</span><span class="w">
    </span><span class="n">pt.size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="w">
  </span><span class="p">)</span><span class="w">
  </span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s1">'figures/basic_canonical_marker_'</span><span class="p">,</span><span class="n">celltype</span><span class="p">,</span><span class="s1">'_vlnPlot.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">3</span><span class="o">*</span><span class="nf">length</span><span class="p">(</span><span class="n">cur_features</span><span class="p">),</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">)</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
  </span><span class="n">dev.off</span><span class="p">()</span><span class="w">

</span><span class="p">}</span></code></pre></figure> </details> <p><strong>Astrocyte marker genes</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Astrocyte_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Pan-neuronal marker genes</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Pan-neuronal_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Excitatory Neuron Markers</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Excitatory-Neuron_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Inhibitory Neuron Markers</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Inhibitory-Neuron_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Microglia Markers</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Microglia_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Oligodendrocyte Markers</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Oligodendrocyte_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p><strong>Oligodendrocyte Progenitor Markers</strong></p> <div class="img"> <img src="/assets/img/tutorials/scRNA1/basic_canonical_marker_Olig.-Progenitor_vlnPlot.png" style="height: 100%; width: 100%; object-fit: contain"> </div> <p>Finally, we can use all of this gene expression information to annotate each cluster with a cell type. Here we use a short hand to name each cluster. For clusters that have high expression of more than one cell type marker gene, we</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">cluster_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
  </span><span class="s1">'0'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ODC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ODC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'3'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ODC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'4'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ASC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'5'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'6'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Doublet'</span><span class="p">,</span><span class="w"> </span><span class="c1"># ASC / Neuron / ODC markers all present,</span><span class="w">
  </span><span class="s1">'7'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'MG'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'8'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'OPC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'9'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'10'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'INH'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'11'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'12'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'INH'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'13'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'14'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'15'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'INH'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'16'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Doublet'</span><span class="p">,</span><span class="w"> </span><span class="c1"># ASC / ODC markers present</span><span class="w">
  </span><span class="s1">'17'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ASC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'18'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'INH'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'19'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'20'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'21'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'22'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'23'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'PER/END'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'24'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ASC'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'25'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'INH'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'26'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'MG'</span><span class="p">,</span><span class="w">
  </span><span class="s1">'27'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'EX'</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># add CellType to seurat metadata</span><span class="w">
</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">CellType</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">cluster_annotations</span><span class="p">[</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">seurat_clusters</span><span class="p">])</span><span class="w">
</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">CellType_cluster</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">CellType</span><span class="p">,</span><span class="w"> </span><span class="s1">'-'</span><span class="p">,</span><span class="w"> </span><span class="n">seurat_obj</span><span class="o">$</span><span class="n">seurat_clusters</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_umap_celltypes.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">DimPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"umap"</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'CellType'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">umap_theme</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'UMAP colored by cell type annotations'</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_umap_celltype_clusters.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">DimPlot</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">reduction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"umap"</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'CellType_cluster'</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">umap_theme</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'UMAP colored by cell type + cluster'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">NoLegend</span><span class="p">()</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_umap_celltypes.png"> </div> </div> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_umap_celltype_clusters.png"> </div> </div> <div class="caption"> </div> <h3 id="identifying-cluster-biomarkers">Identifying cluster biomarkers</h3> <p>In this section, we perform differential gene expression to find cluster marker genes. Marker genes are identified by iteratively comparing gene expression in each cluster to all other clusters. These comparisons can be done using a variety of statistical tests, such as logistic regression. Here we are using <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0844-5" rel="external nofollow noopener" target="_blank">MAST</a>, a hurdle model specifically tailored to the nuances of scRNA-seq data.</p> <p>Seurat provides <code class="language-plaintext highlighter-rouge">FindAllMarkers</code>, a conventient function for iteratively performing these tests for each cluster. Alternatively, we could use the <code class="language-plaintext highlighter-rouge">FindMarkers</code> function to just compare two groups of cells. These functions have a lot of different options that effect the downstream results. Here I will use the default settings, which only look at genes that are up-regulated in the cluster of interest and show non-zero expression in at least 25% of cells in that clusters.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">cluster_markers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FindAllMarkers</span><span class="p">(</span><span class="w">
  </span><span class="n">seurat_obj</span><span class="p">,</span><span class="w">
  </span><span class="n">only.pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">min.pct</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.25</span><span class="p">,</span><span class="w">
  </span><span class="n">logfc.threshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.0</span><span class="p">,</span><span class="w">
  </span><span class="n">method</span><span class="o">=</span><span class="s1">'MAST'</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="n">cluster_markers</span><span class="o">$</span><span class="n">CellType_cluster</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">cluster_annotations</span><span class="p">[</span><span class="n">cluster_markers</span><span class="o">$</span><span class="n">cluster</span><span class="p">]),</span><span class="w"> </span><span class="s1">'-'</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_markers</span><span class="o">$</span><span class="n">cluster</span><span class="p">)</span><span class="w">
</span><span class="n">write.csv</span><span class="p">(</span><span class="n">cluster_markers</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s1">'data/cluster_markers.csv'</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span></code></pre></figure> </details> <p>Now that we have perfomed these tests, we can create some plots that summarize the results. Below we plot the number of DEGs per cluster as a bar plot, and a heatmap of the top 3 DEGs per cluster.</p> <details> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># plot the number of DEGs per cluster:</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">cluster_markers</span><span class="o">$</span><span class="n">CellType_cluster</span><span class="p">)))</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'cluster'</span><span class="p">,</span><span class="w"> </span><span class="s1">'n_DEGs'</span><span class="p">)</span><span class="w">

</span><span class="c1"># bar plot of the number of cells in each sample</span><span class="w">
</span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">n_DEGs</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">reorder</span><span class="p">(</span><span class="n">cluster</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">n_DEGs</span><span class="p">),</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">cluster</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">NoLegend</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">RotatedAxis</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylab</span><span class="p">(</span><span class="nf">expression</span><span class="p">(</span><span class="n">italic</span><span class="p">(</span><span class="n">N</span><span class="p">)[</span><span class="n">DEGs</span><span class="p">]))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlab</span><span class="p">(</span><span class="s1">''</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="w">
    </span><span class="n">panel.grid.minor</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">panel.grid.major.y</span><span class="o">=</span><span class="n">element_line</span><span class="p">(</span><span class="n">colour</span><span class="o">=</span><span class="s2">"lightgray"</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
  </span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_DEGs_barplot.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">300</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot the top 3 DEGs per cluster as a heatmap:</span><span class="w">
</span><span class="n">top_DEGs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cluster_markers</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">group_by</span><span class="p">(</span><span class="n">CellType_cluster</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">top_n</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">wt</span><span class="o">=</span><span class="n">avg_logFC</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">.</span><span class="o">$</span><span class="n">gene</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="s1">'figures/basic_DEGs_heatmap.png'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">300</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s1">'in'</span><span class="p">)</span><span class="w">
</span><span class="n">pdf</span><span class="p">(</span><span class="s1">'figures/basic_DEGs_heatmap.pdf'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">useDingbats</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span><span class="n">DoHeatmap</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">features</span><span class="o">=</span><span class="n">top_DEGs</span><span class="p">,</span><span class="w"> </span><span class="n">group.by</span><span class="o">=</span><span class="s1">'seurat_clusters'</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">scale_fill_gradientn</span><span class="p">(</span><span class="n">colors</span><span class="o">=</span><span class="n">viridis</span><span class="p">(</span><span class="m">256</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">NoLegend</span><span class="p">()</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span></code></pre></figure> </details> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_DEGs_barplot.png"> </div> </div> <div class="caption"> Number of DEGs per cluster </div> <div class="row mt-3"> <div class="col-sm mt-3 mt-md-0"> <img class="img-fluid rounded z-depth-1" src="/assets/img/tutorials/scRNA1/basic_DEGs_heatmap.png"> </div> </div> <div class="caption"> Expression of top 3 DEGs per cluster. Yellow = high, Blue = low. </div> <h3 id="saving-and-loading-seurat-objects">Saving and loading Seurat objects</h3> <p>This concludes the very basics of exploratory data analysis using Seurat. Finally, we will save the processed object so we can use it again later.</p> <details open=""> <summary><b>toggle code</b></summary> <figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1"># save seurat object</span><span class="w">
</span><span class="n">saveRDS</span><span class="p">(</span><span class="n">seurat_obj</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s1">'data/processed_seurat_object.rds'</span><span class="p">)</span><span class="w">

</span><span class="c1"># load seurat object</span><span class="w">
</span><span class="n">seurat_obj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readRDS</span><span class="p">(</span><span class="n">ile</span><span class="o">=</span><span class="s1">'data/processed_seurat_object.rds'</span><span class="p">)</span></code></pre></figure> </details> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Sam Morabito. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?2930004b8d7fcd0a8e00fdcfc8fc9f24"></script> <script defer src="/assets/js/common.js?4a129fbf39254905f505c7246e641eaf"></script> <script defer src="/assets/js/copy_code.js?7254ae07fe9cc5f3a10843e1c0817c9c" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>