<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Alzheimer's disease | Sam Morabito </title> <meta name="author" content="Sam Morabito"> <meta name="description" content="Genomics approaches to study the molecular and cellular aspects of Alzheimer's disease."> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://smorabit.github.io/projects/AD/"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?bf50d6d9dd867d3e0f3b0add94449649"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sam </span> Morabito </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item active"> <a class="nav-link" href="/projects/">projects <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fa-solid fa-moon"></i> <i class="fa-solid fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Alzheimer's disease</h1> <p class="post-description">Genomics approaches to study the molecular and cellular aspects of Alzheimer's disease.</p> </header> <article> <p>Much of my PhD research in the Swarup Lab has focused on studying Alzheimer’s Disease (AD) and neurodegeneration using sequencing approaches like single-cell and spatial -omics. Here I briefly describe each of these projects in chronological order, encompassing my overall PhD journey.</p> <hr> <h3 id="1-integrative-genomics-approach-identifies-conserved-transcriptomic-networks-in-alzheimers-disease">1. Integrative genomics approach identifies conserved transcriptomic networks in Alzheimer’s disease</h3> <div class="row justify-content-sm-center"> <div class="col-sm-8 mt-3 mt-md-0"> In this project, we collected over 1,200 bulk RNA-seq samples from AD brains using three publicly available datasets (ROSMAP, Mt. Sinai Medical, and Mayo Clinic). We performed consensus weighted gene co-expression network analysis (cWGCNA) across these different datasets and brain regions, identifying several gene modules that were altered in disease. This project started out as my rotation project in the Swarup Lab during first year of my PhD, and it eventually became my first first-author publication <a class="citation" href="#Morabito.HMG.2020">(Morabito et al., 2020)</a>. </div> <div class="col-sm-4 mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/consensus_wgcna_net-480.webp 480w,/assets/img/consensus_wgcna_net-800.webp 800w,/assets/img/consensus_wgcna_net-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/consensus_wgcna_net.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <hr> <h3 id="2-single-nucleus-chromatin-accessibility-and-transcriptomic-characterization-of-alzheimers-disease">2. Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease</h3> <div class="row justify-content-sm-center"> <div class="col-sm-4 mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/integrated_umap-480.webp 480w,/assets/img/integrated_umap-800.webp 800w,/assets/img/integrated_umap-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/integrated_umap.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div class="col-sm-8 mt-3 mt-md-0"> This work was the first of its kind to profile the epigenome and transcriptome in AD brains in single cells <a class="citation" href="#Morabito2021">(Morabito* et al., 2021)</a>. In 2019, we began to generate snATAC-seq and snRNA-seq libraries in prefrontal cortex samples of AD donors. We characterized differences in the epigenome and transcriptome between AD and cognitively normal controls for each brain cell type. We used our multi-omic information to link cis-regulatory elements to candidate target genes, and investigated cell-type specific epigenomic landscapes at genetic risk loci. Much of my work on the computational side of this project took place during the onset of the COVID-19 pandemic. </div> </div> <hr> <h3 id="3-spatial-and-single-nucleus-transcriptomic-analysis-of-genetic-and-sporadic-forms-of-alzheimers-disease">3. Spatial and single-nucleus transcriptomic analysis of genetic and sporadic forms of Alzheimer’s Disease</h3> <div class="row justify-content-sm-center"> <div class="col-sm-8 mt-3 mt-md-0"> Here we applied spatial and single-nucleus transcriptomics to compare genetic and sporadic forms of AD <a class="citation" href="#Miyoshi2023">(Miyoshi* et al., 2023)</a>. In particular, we focused comparing Down Syndrome individuals, who generally develop AD with aging (AD in DS), with the sporadic AD population. We also generated spatial data in the 5xFAD mouse model for further comparisons and to identify amyloid-associated gene signatures with aging. We uncovered genes and networks which are similar between AD in DS and sporadic AD, and some which are specific to certain disease stages. Multi-scale network analysis identified an inflammatory glial gene network that we linked to AD polygenic risk and to amyloid accumulation. This project represents the second half of my PhD research. </div> <div class="col-sm-4 mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/spatial_clusters-480.webp 480w,/assets/img/spatial_clusters-800.webp 800w,/assets/img/spatial_clusters-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/spatial_clusters.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <hr> </article> <h2>References</h2> <div class="publications"> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">bioRxiv</abbr> </div> <div id="Miyoshi2023" class="col-sm-8"> <div class="title">Spatial and single-nucleus transcriptomic analysis of genetic and sporadic forms of Alzheimer’s Disease</div> <div class="author"> Emily Miyoshi* ,  <em>Samuel Morabito*</em> ,  Caden M. Henningfield ,  Negin Rahimzadeh ,  Sepideh Kiani Shabestari , and <span class="more-authors" title="click to view 19 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '19 more authors' ? 'Sudeshna Das, Neethu Michael, Fairlie Reese, Zechuan Shi, Zhenkun Cao, Vanessa Scarfone, Miguel A. Arreola, Jackie Lu, Sierra Wright, Justine Silva, Kelsey Leavy, Ira T. Lott, Eric Doran, William H. Yong, Saba Shahin, Mari Perez-Rosendahl, Elizabeth Head, Kim N. Green, Vivek Swarup' : '19 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">19 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2023.07.24.550282v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/miyoshi2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2023.07.24.550282"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.07.24.550282" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The majority of mammalian genes encode multiple transcript isoforms that result from differential promoter use, changes in exonic splicing, and alternative 3′ end choice. Detecting and quantifying transcript isoforms across tissues, cell types, and species has been extremely challenging because transcripts are much longer than the short reads normally used for RNA-seq. By contrast, long-read RNA-seq (LR-RNA-seq) gives the complete structure of most transcripts. We sequenced 264 LR-RNA-seq PacBio libraries totaling over 1 billion circular consensus reads (CCS) for 81 unique human and mouse samples. We detect at least one full-length transcript from 87.7% of annotated human protein coding genes and a total of \textbackslashtextasciitilde200,000 full-length transcripts, \textbackslashtextasciitilde40% of which have novel exon junction chains. To capture and compute on the three sources of transcript structure diversity, we introduce a gene and transcript annotation framework that uses triplets representing the transcript start site, exon junction chain, and transcript end site of each transcript. Using triplets in a simplex representation demonstrates how promoter selection, splice pattern, and 3′ processing are deployed across human tissues, with nearly half of multitranscript protein coding genes showing a clear bias toward one of the three diversity mechanisms. Evaluated across samples, the predominantly expressed transcript changes for 74% of protein coding genes. In evolution, the human and mouse transcriptomes are globally similar in types of transcript structure diversity, yet among individual orthologous gene pairs, more than half (57.8%) show substantial differences in mechanism of diversification in matching tissues. This initial large-scale survey of human and mouse long-read transcriptomes provides a foundation for further analyses of alternative transcript usage, and is complemented by short-read and microRNA data on the same samples and by epigenome data elsewhere in the ENCODE4 collection.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/natgen_july2021-480.webp 480w,/assets/img/publication_preview/natgen_july2021-800.webp 800w,/assets/img/publication_preview/natgen_july2021-1400.webp 1400w," sizes="200px" type="image/webp"> <img src="/assets/img/publication_preview/natgen_july2021.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="natgen_july2021.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> </div> <div id="Morabito2021" class="col-sm-8"> <div class="title">Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito*</em> ,  Emily Miyoshi* ,  Neethu Michael ,  Saba Shahin ,  Alessandra Cadete Martini , and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Elizabeth Head, Justine Silva, Kelsey Leavy, Mari Perez-Rosendahl, Vivek Swarup' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">5 more authors</span> </div> <div class="periodical"> <em>Nature Genetics</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41588-021-00894-z" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/morabito_NatGen_2021.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41588-021-00894-z"></span> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41588-021-00894-z" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The gene-regulatory landscape of the brain is highly dynamic in health and disease, coordinating a menagerie of biological processes across distinct cell types. Here, we present a multi-omic single-nucleus study of 191,890 nuclei in late-stage Alzheimer’s disease (AD), accessible through our web portal, profiling chromatin accessibility and gene expression in the same biological samples and uncovering vast cellular heterogeneity. We identified cell-type-specific, disease-associated candidate cis-regulatory elements and their candidate target genes, including an oligodendrocyte-associated regulatory module containing links to APOE and CLU. We describe cis-regulatory relationships in specific cell types at a subset of AD risk loci defined by genome-wide association studies, demonstrating the utility of this multi-omic single-nucleus approach. Trajectory analysis of glial populations identified disease-relevant transcription factors, such as SREBF1, and their regulatory targets. Finally, we introduce single-nucleus consensus weighted gene coexpression analysis, a coexpression network analysis strategy robust to sparse single-cell data, and perform a systems-level analysis of the AD transcriptome.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/hmg_2020-480.webp 480w,/assets/img/publication_preview/hmg_2020-800.webp 800w,/assets/img/publication_preview/hmg_2020-1400.webp 1400w," sizes="200px" type="image/webp"> <img src="/assets/img/publication_preview/hmg_2020.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="hmg_2020.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> </div> <div id="Morabito.HMG.2020" class="col-sm-8"> <div class="title">Integrative genomics approach identifies conserved transcriptomic networks in Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito</em> ,  Emily Miyoshi ,  Neethu Michael ,  and  Vivek Swarup </div> <div class="periodical"> <em>Human Molecular Genetics</em>, Aug 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1093/hmg/ddaa182" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/morabito-HMG-2020.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="/assets/pdf/morabito-HMG-2020-supp.pdf" class="btn btn-sm z-depth-0" role="button">Supp</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1093/hmg/ddaa182" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Alzheimer’s disease (AD) is a devastating neurological disorder characterized by changes in cell-type proportions and consequently marked alterations of the transcriptome. Here we use a data-driven systems biology meta-analytical approach across three human ad cohorts, encompassing six cortical brain regions, and integrate with multi-scale datasets comprising of DNA methylation, histone acetylation, transcriptome- and genome-wide association studies, and quantitative trait loci to further characterize the genetic architecture of ad. We perform co-expression network analysis across more than twelve hundred human brain samples, identifying robust ad-associated dysregulation of the transcriptome, unaltered in normal human aging. We assess the cell-type specificity of ad gene co-expression changes and estimate cell-type proportion changes in human ad by integrating co-expression modules with single-cell transcriptome data generated from 27 321 nuclei from human postmortem prefrontal cortical tissue. We also show that genetic variants of ad are enriched in a microglial ad-associated module and identify key transcription factors regulating co-expressed modules. Additionally, we validate our results in multiple published human ad gene expression datasets, which can be easily accessed using our online resource (https://swaruplab.bio.uci.edu/consensusAD).</p> </div> </div> </div> </li></ol> </div> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Sam Morabito. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?2930004b8d7fcd0a8e00fdcfc8fc9f24"></script> <script defer src="/assets/js/common.js?4a129fbf39254905f505c7246e641eaf"></script> <script defer src="/assets/js/copy_code.js?7254ae07fe9cc5f3a10843e1c0817c9c" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>