<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> publications | Sam Morabito </title> <meta name="author" content="Sam Morabito"> <meta name="description" content=""> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://smorabit.github.io/publications/"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?bf50d6d9dd867d3e0f3b0add94449649"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sam </span> Morabito </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fa-solid fa-moon"></i> <i class="fa-solid fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description"></p> </header> <article> <p><b> *These authors contributed equally to this work </b></p> <div class="publications"> <h2 class="bibliography">2024</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">Cell Rep.</abbr> </div> <div id="Childs2024" class="col-sm-8"> <div class="title">Relapse to cocaine seeking is regulated by medial habenula NR4A2/NURR1 in mice</div> <div class="author"> Jessica E. Childs* ,  <em>Samuel Morabito*</em> ,  Sudeshna Das ,  Caterina Santelli ,  Victoria Pham , and <span class="more-authors" title="click to view 7 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '7 more authors' ? 'Kelly Kusche, Vanessa Alizo Vera, Fairlie Reese, Rianne R. Campbell, Dina P. Matheos, Vivek Swarup, Marcelo A. Wood' : '7 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">7 more authors</span> </div> <div class="periodical"> <em>Cell Reports</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S2211124724002845" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/childs2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.celrep.2024.113956"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.celrep.2024.113956" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Drugs of abuse can persistently change the reward circuit in ways that contribute to relapse behavior, partly via mechanisms that regulate chromatin structure and function. Nuclear orphan receptor subfamily4 groupA member2 (NR4A2, also known as NURR1) is an important effector of histone deacetylase 3 (HDAC3)-dependent mechanisms in persistent memory processes and is highly expressed in the medial habenula (MHb), a region that regulates nicotine-associated behaviors. Here, expressing the Nr4a2 dominant negative (Nurr2c) in the MHb blocks reinstatement of cocaine seeking in mice. We use single-nucleus transcriptomics to characterize the molecular cascade following Nr4a2 manipulation, revealing changes in transcriptional networks related to addiction, neuroplasticity, and GABAergic and glutamatergic signaling. The network controlled by NR4A2 is characterized using a transcription factor regulatory network inference algorithm. These results identify the MHb as a pivotal regulator of relapse behavior and demonstrate the importance of NR4A2 as a key mechanism driving the MHb component of relapse.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">DADM</abbr> </div> <div id="Agudo2024" class="col-sm-8"> <div class="title">BIN1K358R suppresses glial response to plaques in mouse model of Alzheimer’s disease</div> <div class="author"> Laura Fernandez Garcia‐Agudo ,  Zechuan Shi ,  Ian F. Smith ,  Enikö A. Kramár ,  Katelynn Tran , and <span class="more-authors" title="click to view 22 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '22 more authors' ? 'Shimako Kawauchi, Shuling Wang, Sherilyn Collins, Amber Walker, Kai‐Xuan Shi, Jonathan Neumann, Heidi Yahan Liang, Celia Da Cunha, Giedre Milinkeviciute, Samuel Morabito, Emily Miyoshi, Narges Rezaie, Angela Gomez‐Arboledas, Adrian Mendoza Arvilla, Daryan Iman Ghaemi, Andrea J. Tenner, Frank M. LaFerla, Marcelo A. Wood, Ali Mortazavi, Vivek Swarup, Grant R. MacGregor, Kim N. Green' : '22 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">22 more authors</span> </div> <div class="periodical"> <em>Alzheimer’s &amp; Dementia</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://alz-journals.onlinelibrary.wiley.com/doi/full/10.1002/alz.13767" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Agudo2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1002/alz.13767"></span> <span class="__dimensions_badge_embed__" data-doi="10.1002/alz.13767" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The BIN1 coding variant rs138047593 (K358R) is linked to Late‐Onset Alzheimer’s Disease (LOAD) via targeted exome sequencing. To elucidate the functional consequences of this rare coding variant on brain amyloidosis and neuroinflammation, we generated BIN1K358R knock‐in mice using CRISPR/Cas9 technology. These mice were subsequently bred with 5xFAD transgenic mice, which serve as a model for Alzheimer’s pathology. The presence of the BIN1K358R variant leads to increased cerebral amyloid deposition, with a dampened response of astrocytes and oligodendrocytes, but not microglia, at both the cellular and transcriptional levels. This correlates with decreased neurofilament light chain in both plasma and brain tissue. Synaptic densities are significantly increased in both wild‐type and 5xFAD backgrounds homozygous for the BIN1K358R variant. The BIN1 K358R variant modulates amyloid pathology in 5xFAD mice, attenuates the astrocytic and oligodendrocytic responses to amyloid plaques, decreases damage markers, and elevates synaptic densities. BIN1 rs138047593 (K358R) coding variant is associated with increased risk of LOAD. BIN1 K358R variant increases amyloid plaque load in 12‐month‐old 5xFAD mice. BIN1 K358R variant dampens astrocytic and oligodendrocytic response to plaques. BIN1 K358R variant decreases neuronal damage in 5xFAD mice. BIN1 K358R upregulates synaptic densities and modulates synaptic transmission.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">bioRxiv</abbr> </div> <div id="Miyoshi2023" class="col-sm-8"> <div class="title">Spatial and single-nucleus transcriptomic analysis of genetic and sporadic forms of Alzheimer’s Disease</div> <div class="author"> Emily Miyoshi* ,  <em>Samuel Morabito*</em> ,  Caden M. Henningfield ,  Negin Rahimzadeh ,  Sepideh Kiani Shabestari , and <span class="more-authors" title="click to view 19 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '19 more authors' ? 'Sudeshna Das, Neethu Michael, Fairlie Reese, Zechuan Shi, Zhenkun Cao, Vanessa Scarfone, Miguel A. Arreola, Jackie Lu, Sierra Wright, Justine Silva, Kelsey Leavy, Ira T. Lott, Eric Doran, William H. Yong, Saba Shahin, Mari Perez-Rosendahl, Elizabeth Head, Kim N. Green, Vivek Swarup' : '19 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">19 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2023.07.24.550282v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/miyoshi2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2023.07.24.550282"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.07.24.550282" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The majority of mammalian genes encode multiple transcript isoforms that result from differential promoter use, changes in exonic splicing, and alternative 3′ end choice. Detecting and quantifying transcript isoforms across tissues, cell types, and species has been extremely challenging because transcripts are much longer than the short reads normally used for RNA-seq. By contrast, long-read RNA-seq (LR-RNA-seq) gives the complete structure of most transcripts. We sequenced 264 LR-RNA-seq PacBio libraries totaling over 1 billion circular consensus reads (CCS) for 81 unique human and mouse samples. We detect at least one full-length transcript from 87.7% of annotated human protein coding genes and a total of \textbackslashtextasciitilde200,000 full-length transcripts, \textbackslashtextasciitilde40% of which have novel exon junction chains. To capture and compute on the three sources of transcript structure diversity, we introduce a gene and transcript annotation framework that uses triplets representing the transcript start site, exon junction chain, and transcript end site of each transcript. Using triplets in a simplex representation demonstrates how promoter selection, splice pattern, and 3′ processing are deployed across human tissues, with nearly half of multitranscript protein coding genes showing a clear bias toward one of the three diversity mechanisms. Evaluated across samples, the predominantly expressed transcript changes for 74% of protein coding genes. In evolution, the human and mouse transcriptomes are globally similar in types of transcript structure diversity, yet among individual orthologous gene pairs, more than half (57.8%) show substantial differences in mechanism of diversification in matching tissues. This initial large-scale survey of human and mouse long-read transcriptomes provides a foundation for further analyses of alternative transcript usage, and is complemented by short-read and microRNA data on the same samples and by epigenome data elsewhere in the ENCODE4 collection.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cellrepmeth_2023-480.webp 480w,/assets/img/publication_preview/cellrepmeth_2023-800.webp 800w,/assets/img/publication_preview/cellrepmeth_2023-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/cellrepmeth_2023.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cellrepmeth_2023.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito2023" class="col-sm-8"> <div class="title">hdWGCNA identifies co-expression networks in high-dimensional transcriptomics data</div> <div class="author"> <em>Samuel Morabito</em> ,  Fairlie Reese ,  Negin Rahimzadeh ,  Emily Miyoshi ,  and  Vivek Swarup </div> <div class="periodical"> <em>Cell Reports Methods</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(23)00127-3" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/morabito_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.crmeth.2023.100498"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.crmeth.2023.100498" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Biological systems are immensely complex, organized into a multi-scale hierarchy of functional units based on tightly regulated interactions between distinct molecules, cells, organs, and organisms. While experimental methods enable transcriptome-wide measurements across millions of cells, popular bioinformatic tools do not support systems-level analysis. Here we present hdWGCNA, a comprehensive framework for analyzing co-expression networks in high-dimensional transcriptomics data such as single-cell and spatial RNA sequencing (RNA-seq). hdWGCNA provides functions for network inference, gene module identification, gene enrichment analysis, statistical tests, and data visualization. Beyond conventional single-cell RNA-seq, hdWGCNA is capable of performing isoform-level network analysis using long-read single-cell data. We showcase hdWGCNA using data from autism spectrum disorder and Alzheimer’s disease brain samples, identifying disease-relevant co-expression network modules. hdWGCNA is directly compatible with Seurat, a widely used R package for single-cell and spatial transcriptomics analysis, and we demonstrate the scalability of hdWGCNA by analyzing a dataset containing nearly 1 million cells</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">bioRxiv</abbr> </div> <div id="Reese2023" class="col-sm-8"> <div class="title">The ENCODE4 long-read RNA-seq collection reveals distinct classes of transcript structure diversity</div> <div class="author"> Fairlie Reese ,  Brian Williams ,  Gabriela Balderrama-Gutierrez ,  Dana Wyman ,  Muhammed Hasan Çelik , and <span class="more-authors" title="click to view 40 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '40 more authors' ? 'Elisabeth Rebboah, Narges Rezaie, Diane Trout, Milad Razavi-Mohseni, Yunzhe Jiang, Beatrice Borsari, Samuel Morabito, Heidi Yahan Liang, Cassandra J McGill, Sorena Rahmanian, Jasmine Sakr, Shan Jiang, Weihua Zeng, Klebea Carvalho, Annika K Weimer, Louise A Dionne, Ariel McShane, Karan Bedi, Shaimae I Elhajjajy, Sean Upchurch, Jennifer Jou, Ingrid Youngworth, Idan Gabdank, Paul Sud, Otto Jolanki, J Seth Strattan, Meenakshi S Kagda, Michael P Snyder, Ben C Hitz, Jill E Moore, Zhiping Weng, David Bennett, Laura Reinholdt, Mats Ljungman, Michael A Beer, Mark B Gerstein, Lior Pachter, Roderic Guigó, Barbara J Wold, Ali Mortazavi' : '40 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">40 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2023.05.15.540865v1.abstract" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/reese2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2023.05.15.540865"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.05.15.540865" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The majority of mammalian genes encode multiple transcript isoforms that result from differential promoter use, changes in exonic splicing, and alternative 3′ end choice. Detecting and quantifying transcript isoforms across tissues, cell types, and species has been extremely challenging because transcripts are much longer than the short reads normally used for RNA-seq. By contrast, long-read RNA-seq (LR-RNA-seq) gives the complete structure of most transcripts. We sequenced 264 LR-RNA-seq PacBio libraries totaling over 1 billion circular consensus reads (CCS) for 81 unique human and mouse samples. We detect at least one full-length transcript from 87.7% of annotated human protein coding genes and a total of \textbackslashtextasciitilde200,000 full-length transcripts, \textbackslashtextasciitilde40% of which have novel exon junction chains. To capture and compute on the three sources of transcript structure diversity, we introduce a gene and transcript annotation framework that uses triplets representing the transcript start site, exon junction chain, and transcript end site of each transcript. Using triplets in a simplex representation demonstrates how promoter selection, splice pattern, and 3′ processing are deployed across human tissues, with nearly half of multitranscript protein coding genes showing a clear bias toward one of the three diversity mechanisms. Evaluated across samples, the predominantly expressed transcript changes for 74% of protein coding genes. In evolution, the human and mouse transcriptomes are globally similar in types of transcript structure diversity, yet among individual orthologous gene pairs, more than half (57.8%) show substantial differences in mechanism of diversification in matching tissues. This initial large-scale survey of human and mouse long-read transcriptomes provides a foundation for further analyses of alternative transcript usage, and is complemented by short-read and microRNA data on the same samples and by epigenome data elsewhere in the ENCODE4 collection.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cellmetab_2023-480.webp 480w,/assets/img/publication_preview/cellmetab_2023-800.webp 800w,/assets/img/publication_preview/cellmetab_2023-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/cellmetab_2023.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cellmetab_2023.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Costanzo.2023.Cell Metabolism" class="col-sm-8"> <div class="title">The Type 2 Diabetes Knowledge Portal: An open access genetic resource dedicated to type 2 diabetes and related traits</div> <div class="author"> Maria C. Costanzo ,  Marcin von Grotthuss ,  Jeffrey Massung ,  Dongkeun Jang ,  Lizz Caulkins , and <span class="more-authors" title="click to view 187 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '187 more authors' ? 'Ryan Koesterer, Clint Gilbert, Ryan P. Welch, Parul Kudtarkar, Quy Hoang, Andrew P. Boughton, Preeti Singh, Ying Sun, Marc Duby, Annie Moriondo, Trang Nguyen, Patrick Smadbeck, Benjamin R. Alexander, MacKenzie Brandes, Mary Carmichael, Peter Dornbos, Todd Green, Kenneth C. Huellas-Bruskiewicz, Yue Ji, Alexandria Kluge, Aoife C. McMahon, Josep M. Mercader, Oliver Ruebenacker, Sebanti Sengupta, Dylan Spalding, Daniel Taliun, AMP-T2D Consortium, Gonçalo Abecasis, Beena Akolkar, Benjamin R. Alexander, Nicholette D. Allred, David Altshuler, Jennifer E. Below, Richard Bergman, Joline W.J. Beulens, John Blangero, Michael Boehnke, Krister Bokvist, Erwin Bottinger, Andrew P. Boughton, Donald Bowden, M. Julia Brosnan, Christopher Brown, Kenneth Bruskiewicz, Noël P. Burtt, Mary Carmichael, Lizz Caulkins, Inês Cebola, John Chambers, Yii-Der Ida Chen, Andriy Cherkas, Audrey Y. Chu, Christopher Clark, Melina Claussnitzer, Maria C. Costanzo, Nancy J. Cox, Marcel den Hoed, Duc Dong, Marc Duby, Ravindranath Duggirala, Josée Dupuis, Petra J.M. Elders, Jesse M. Engreitz, Eric Fauman, Jorge Ferrer, Jason Flannick, Paul Flicek, Matthew Flickinger, Jose C. Florez, Caroline S. Fox, Timothy M. Frayling, Kelly A. Frazer, Kyle J. Gaulton, Clint Gilbert, Anna L. Gloyn, Todd Green, Craig L. Hanis, Robert Hanson, Andrew T. Hattersley, Quy Hoang, Hae Kyung Im, Sidra Iqbal, Suzanne B.R. Jacobs, Dong-Keun Jang, Tad Jordan, Tania Kamphaus, Fredrik Karpe, Thomas M. Keane, Seung K. Kim, Alexandria Kluge, Ryan Koesterer, Parul Kudtarkar, Kasper Lage, Leslie A. Lange, Mitchell Lazar, Donna Lehman, Ching-Ti Liu, Ruth J.F. Loos, Ronald Ching-wan Ma, Patrick MacDonald, Jeffrey Massung, Matthew T. Maurano, Mark I. McCarthy, Gil McVean, James B. Meigs, Josep M. Mercader, Melissa R. Miller, Braxton Mitchell, Karen L. Mohlke, Samuel Morabito, Claire Morgan, Shannon Mullican, Sharvari Narendra, Maggie C.Y. Ng, Lynette Nguyen, Colin N.A. Palmer, Stephen C.J. Parker, Antonio Parrado, Afshin Parsa, Aaron C. Pawlyk, Ewan R. Pearson, Andrew Plump, Michael Province, Thomas Quertermous, Susan Redline, Dermot F. Reilly, Bing Ren, Stephen S. Rich, J. Brent Richards, Jerome I. Rotter, Oliver Ruebenacker, Hartmut Ruetten, Rany M. Salem, Maike Sander, Michael Sanders, Dharambir Sanghera, Laura J. Scott, Sebanti Sengupta, David Siedzik, Xueling Sim, Preeti Singh, Robert Sladek, Kerrin Small, Philip Smith, Peter Stein, Dylan Spalding, Heather M. Stringham, Ying Sun, Katalin Susztak, Leen M. ’t Hart, Daniel Taliun, Kent Taylor, Melissa K. Thomas, Jennifer A. Todd, Miriam S. Udler, Benjamin Voight, Marcin von Grotthuss, Andre Wan, Ryan P. Welch, David Wholley, Kaan Yuksel, Norann A. Zaghloul, Philip Smith, Melissa K. Thomas, Beena Akolkar, M. Julia Brosnan, Andriy Cherkas, Audrey Y. Chu, Eric B. Fauman, Caroline S. Fox, Tania Nayak Kamphaus, Melissa R. Miller, Lynette Nguyen, Afshin Parsa, Dermot F. Reilly, Hartmut Ruetten, David Wholley, Norann A. Zaghloul, Gonçalo R. Abecasis, David Altshuler, Thomas M. Keane, Mark I. McCarthy, Kyle J. Gaulton, Jose C. Florez, Michael Boehnke, Noël P. Burtt, Jason Flannick' : '187 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">187 more authors</span> </div> <div class="periodical"> <em>Cell Metabolism</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.cell.com/cell-metabolism/fulltext/S1550-4131(23)00079-7" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/costanzo_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.cmet.2023.03.001"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.cmet.2023.03.001" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Associations between human genetic variation and clinical phenotypes have become a foundation of biomedical research. Most repositories of these data seek to be disease-agnostic and therefore lack disease-focused views. The Type 2 Diabetes Knowledge Portal (T2DKP) is a public resource of genetic datasets and genomic annotations dedicated to type 2 diabetes (T2D) and related traits. Here, we seek to make the T2DKP more accessible to prospective users and more useful to existing users. First, we evaluate the T2DKP’s comprehensiveness by comparing its datasets with those of other repositories. Second, we describe how researchers unfamiliar with human genetic data can begin using and correctly interpreting them via the T2DKP. Third, we describe how existing users can extend their current workflows to use the full suite of tools offered by the T2DKP. We finally discuss the lessons offered by the T2DKP toward the goal of democratizing access to complex disease genetic results.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/neuron_2022-480.webp 480w,/assets/img/publication_preview/neuron_2022-800.webp 800w,/assets/img/publication_preview/neuron_2022-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/neuron_2022.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="neuron_2022.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="OteroGarcia2022" class="col-sm-8"> <div class="title">Molecular signatures underlying neurofibrillary tangle susceptibility in Alzheimer’s disease</div> <div class="author"> Marcos Otero-Garcia ,  Sameehan U. Mahajani ,  Debia Wakhloo ,  Weijing Tang ,  Yue-Qiang Xue , and <span class="more-authors" title="click to view 12 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '12 more authors' ? 'Samuel Morabito, Jie Pan, Jane Oberhauser, Angela E. Madira, Tamara Shakouri, Yongning Deng, Thomas Allison, Zihuai He, William E. Lowry, Riki Kawaguchi, Vivek Swarup, Inma Cobos' : '12 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">12 more authors</span> </div> <div class="periodical"> <em>Neuron</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.cell.com/neuron/fulltext/S0896-6273(22)00600-6?rss=yes" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/otero_garcia_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.neuron.2022.06.021"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.neuron.2022.06.021" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Tau aggregation in neurofibrillary tangles (NFTs) is closely associated with neurodegeneration and cognitive decline in Alzheimer’s disease (AD). However, the molecular signatures that distinguish between aggregation-prone and aggregation-resistant cell states are unknown. We developed methods for the high-throughput isolation and transcriptome profiling of single somas with NFTs from the human AD brain, quantified the susceptibility of 20 neocortical subtypes for NFT formation and death, and identified both shared and cell-type-specific signatures. NFT-bearing neurons shared a marked upregulation of synaptic transmission-related genes, including a core set of 63 genes enriched for synaptic vesicle cycling. Oxidative phosphorylation and mitochondrial dysfunction were highly cell-type dependent. Apoptosis was only modestly enriched, and the susceptibilities of NFT-bearing and NFT-free neurons for death were highly similar. Our analysis suggests that NFTs represent cell-type-specific responses to stress and synaptic dysfunction. We provide a resource for biomarker discovery and the investigation of tau-dependent and tau-independent mechanisms of neurodegeneration.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cellrep_2022-480.webp 480w,/assets/img/publication_preview/cellrep_2022-800.webp 800w,/assets/img/publication_preview/cellrep_2022-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/cellrep_2022.gif" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cellrep_2022.gif" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Shabestari2022" class="col-sm-8"> <div class="title">Absence of microglia promotes diverse pathologies and early lethality in Alzheimer’s disease mice</div> <div class="author"> Sepideh Kiani Shabestari ,  <em>Samuel Morabito</em> ,  Emma Pascal Danhash ,  Amanda McQuade ,  Jessica Ramirez Sanchez , and <span class="more-authors" title="click to view 16 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '16 more authors' ? 'Emily Miyoshi, Jean Paul Chadarevian, Christel Claes, Morgan Alexandra Coburn, Jonathan Hasselmann, Jorge Hidalgo, Kayla Nhi Tran, Alessandra C. Martini, Winston Chang Rothermich, Jesse Pascual, Elizabeth Head, David A. Hume, Clare Pridans, Hayk Davtyan, Vivek Swarup, Mathew Blurton-Jones' : '16 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">16 more authors</span> </div> <div class="periodical"> <em>Cell Reports</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.celrep.2022.110961" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/shabestari_cr2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.celrep.2022.110961"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.celrep.2022.110961" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Microglia are strongly implicated in the development and progression of Alzheimer’s disease (AD), yet their impact on pathology and lifespan remains unclear. Here we utilize a CSF1R hypomorphic mouse to generate a model of AD that genetically lacks microglia. The resulting microglial-deficient mice exhibit a profound shift from parenchymal amyloid plaques to cerebral amyloid angiopathy (CAA), which is accompanied by numerous transcriptional changes, greatly increased brain calcification and hemorrhages, and premature lethality. Remarkably, a single injection of wild-type microglia into adult mice repopulates the microglial niche and prevents each of these pathological changes. Taken together, these results indicate the protective functions of microglia in reducing CAA, blood-brain barrier dysfunction, and brain calcification. To further understand the clinical implications of these findings, human AD tissue and iPSC-microglia were examined, providing evidence that microglia phagocytose calcium crystals, and this process is impaired by loss of the AD risk gene, TREM2.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/starprotoc_2022-480.webp 480w,/assets/img/publication_preview/starprotoc_2022-800.webp 800w,/assets/img/publication_preview/starprotoc_2022-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/starprotoc_2022.gif" class="preview z-depth-1 rounded" width="100%" height="auto" alt="starprotoc_2022.gif" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="10.1016/j.xpro.2022.101491" class="col-sm-8"> <div class="title">Protocol for single-nucleus ATAC sequencing and bioinformatic analysis in frozen human brain tissue</div> <div class="author"> Zechuan Shi* ,  Sudeshna Das* ,  <em>Samuel Morabito</em> ,  Emily Miyoshi ,  and  Vivek Swarup </div> <div class="periodical"> <em>STAR Protocols</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S2666166722003719" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/shi_star_protocol_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.xpro.2022.101491" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Single-nucleus ATAC sequencing (snATAC-seq) employs a hyperactive Tn5 transposase to gain precise information about the cis-regulatory elements in specific cell types. However, the standard protocol of snATAC-seq is not optimized for all tissues, including the brain. Here, we present a modified protocol for single-nuclei isolation from postmortem frozen human brain tissue, followed by snATAC-seq library preparation and sequencing. We also describe an integrated bioinformatics analysis pipeline using an R package (ArchRtoSignac) to robustly analyze snATAC-seq data. For complete details on the use and execution of this protocol, please refer to Morabito et al. (2021).</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">Sci Rep</abbr> </div> <div id="Barahona2022" class="col-sm-8"> <div class="title">Cortical diurnal rhythms remain intact with microglial depletion</div> <div class="author"> Rocio A. Barahona ,  <em>Samuel Morabito</em> ,  Vivek Swarup ,  and  Kim N. Green </div> <div class="periodical"> <em>Scientific Reports</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41598-021-04079-w" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/barahona_SciRep_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41598-021-04079-w" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Microglia are subject to change in tandem with the endogenously generated biological oscillations known as our circadian rhythm. Studies have shown microglia harbor an intrinsic molecular clock which regulates diurnal changes in morphology and influences inflammatory responses. In the adult brain, microglia play an important role in the regulation of condensed extracellular matrix structures called perineuronal nets (PNNs), and it has been suggested that PNNs are also regulated in a circadian and diurnal manner. We sought to determine whether microglia mediate the diurnal regulation of PNNs via CSF1R inhibitor dependent microglial depletion in C57BL/6J mice, and how the absence of microglia might affect cortical diurnal gene expression rhythms. While we observe diurnal differences in microglial morphology, where microglia are most ramified at the onset of the dark phase, we do not find diurnal differences in PNN intensity. However, PNN intensity increases across many brain regions in the absence of microglia, supporting a role for microglia in the regulation of PNNs. Here, we also show that cortical diurnal gene expression rhythms are intact, with no cycling gene changes without microglia. These findings demonstrate a role for microglia in the maintenance of PNNs, but not in the maintenance of diurnal rhythms.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/natgen_july2021-480.webp 480w,/assets/img/publication_preview/natgen_july2021-800.webp 800w,/assets/img/publication_preview/natgen_july2021-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/natgen_july2021.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="natgen_july2021.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito2021" class="col-sm-8"> <div class="title">Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito*</em> ,  Emily Miyoshi* ,  Neethu Michael ,  Saba Shahin ,  Alessandra Cadete Martini , and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Elizabeth Head, Justine Silva, Kelsey Leavy, Mari Perez-Rosendahl, Vivek Swarup' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">5 more authors</span> </div> <div class="periodical"> <em>Nature Genetics</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41588-021-00894-z" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/morabito_NatGen_2021.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41588-021-00894-z"></span> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41588-021-00894-z" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The gene-regulatory landscape of the brain is highly dynamic in health and disease, coordinating a menagerie of biological processes across distinct cell types. Here, we present a multi-omic single-nucleus study of 191,890 nuclei in late-stage Alzheimer’s disease (AD), accessible through our web portal, profiling chromatin accessibility and gene expression in the same biological samples and uncovering vast cellular heterogeneity. We identified cell-type-specific, disease-associated candidate cis-regulatory elements and their candidate target genes, including an oligodendrocyte-associated regulatory module containing links to APOE and CLU. We describe cis-regulatory relationships in specific cell types at a subset of AD risk loci defined by genome-wide association studies, demonstrating the utility of this multi-omic single-nucleus approach. Trajectory analysis of glial populations identified disease-relevant transcription factors, such as SREBF1, and their regulatory targets. Finally, we introduce single-nucleus consensus weighted gene coexpression analysis, a coexpression network analysis strategy robust to sparse single-cell data, and perform a systems-level analysis of the AD transcriptome.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/nbd_2022-480.webp 480w,/assets/img/publication_preview/nbd_2022-800.webp 800w,/assets/img/publication_preview/nbd_2022-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/nbd_2022.gif" class="preview z-depth-1 rounded" width="100%" height="auto" alt="nbd_2022.gif" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Miyoshi.NBD.2021" class="col-sm-8"> <div class="title">Systems biology approaches to unravel the molecular and genetic architecture of Alzheimer’s disease and related tauopathies</div> <div class="author"> Emily Miyoshi* ,  <em>Samuel Morabito*</em> ,  and  Vivek Swarup </div> <div class="periodical"> <em>Neurobiology of Disease</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.nbd.2021.105530" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/miyoshi_nbd_2021.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.nbd.2021.105530" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Over the years, genetic studies have identified multiple genetic risk variants associated with neurodegenerative disorders and helped reveal new biological pathways and genes of interest. However, genetic risk variants commonly reside in non-coding regions and may regulate distant genes rather than the nearest gene, as well as a gene’s interaction partners in biological networks. Systems biology and functional genomics approaches provide the framework to unravel the functional significance of genetic risk variants in disease. In this review, we summarize the genetic and transcriptomic studies of Alzheimer’s disease and related tauopathies and focus on the advantages of performing systems-level analyses to interrogate the biological pathways underlying neurodegeneration. Finally, we highlight new avenues of multi-omics analysis with single-cell approaches, which provides unparalleled opportunities to systematically explore cellular heterogeneity, and present an example of how to integrate publicly available single-cell datasets. Systems-level analysis has illuminated the function of many disease risk genes, but much work remains to study tauopathies and to understand spatiotemporal gene expression changes of specific cell types.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/hmg_2020-480.webp 480w,/assets/img/publication_preview/hmg_2020-800.webp 800w,/assets/img/publication_preview/hmg_2020-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/hmg_2020.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="hmg_2020.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito.HMG.2020" class="col-sm-8"> <div class="title">Integrative genomics approach identifies conserved transcriptomic networks in Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito</em> ,  Emily Miyoshi ,  Neethu Michael ,  and  Vivek Swarup </div> <div class="periodical"> <em>Human Molecular Genetics</em>, Aug 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1093/hmg/ddaa182" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/morabito-HMG-2020.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="/assets/pdf/morabito-HMG-2020-supp.pdf" class="btn btn-sm z-depth-0" role="button">Supp</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1093/hmg/ddaa182" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Alzheimer’s disease (AD) is a devastating neurological disorder characterized by changes in cell-type proportions and consequently marked alterations of the transcriptome. Here we use a data-driven systems biology meta-analytical approach across three human ad cohorts, encompassing six cortical brain regions, and integrate with multi-scale datasets comprising of DNA methylation, histone acetylation, transcriptome- and genome-wide association studies, and quantitative trait loci to further characterize the genetic architecture of ad. We perform co-expression network analysis across more than twelve hundred human brain samples, identifying robust ad-associated dysregulation of the transcriptome, unaltered in normal human aging. We assess the cell-type specificity of ad gene co-expression changes and estimate cell-type proportion changes in human ad by integrating co-expression modules with single-cell transcriptome data generated from 27 321 nuclei from human postmortem prefrontal cortical tissue. We also show that genetic variants of ad are enriched in a microglial ad-associated module and identify key transcription factors regulating co-expressed modules. Additionally, we validate our results in multiple published human ad gene expression datasets, which can be easily accessed using our online resource (https://swaruplab.bio.uci.edu/consensusAD).</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">Comms Bio</abbr> </div> <div id="Chen2019" class="col-sm-8"> <div class="title">Single-cell landscape in mammary epithelium reveals bipotent-like cells associated with breast cancer risk and outcome</div> <div class="author"> Weiyan Chen ,  Samuel J. Morabito ,  Kai Kessenbrock ,  Tariq Enver ,  Kerstin B. Meyer , and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Andrew E. Teschendorff' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '3'); ">1 more author</span> </div> <div class="periodical"> <em>Communications Biology</em>, Dec 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://www.nature.com/articles/s42003-019-0554-8" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/commbio_2019.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1038/s42003-019-0554-8" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Adult stem-cells may serve as the cell-of-origin for cancer, yet their unbiased identification in single cell RNA sequencing data is challenging due to the high dropout rate. In the case of breast, the existence of a bipotent stem-like state is also controversial. Here we apply a marker-free algorithm to scRNA-Seq data from the human mammary epithelium, revealing a high-potency cell-state enriched for an independent mammary stem-cell expression module. We validate this stem-like state in independent scRNA-Seq data. Our algorithm further predicts that the stem-like state is bipotent, a prediction we are able to validate using FACS sorted bulk expression data. The bipotent stem-like state correlates with clinical outcome in basal breast cancer and is characterized by overexpression of YBX1 and ENO1, two modulators of basal breast cancer risk. This study illustrates the power of a marker-free computational framework to identify a novel bipotent stem-like state in the mammary epithelium.</p> </div> </div> </div> </li></ol> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Sam Morabito. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?2930004b8d7fcd0a8e00fdcfc8fc9f24"></script> <script defer src="/assets/js/common.js?4a129fbf39254905f505c7246e641eaf"></script> <script defer src="/assets/js/copy_code.js?7254ae07fe9cc5f3a10843e1c0817c9c" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>