<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> publications | Sam Morabito </title> <meta name="author" content="Sam Morabito"> <meta name="description" content=""> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@300;400;500;600;700&amp;family=Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://smorabit.github.io/publications/"> <script src="/assets/js/theme.js?a81d82887dd692e91686b43de4542f18"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sam</span> Morabito </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/photos/">photos </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description"></p> </header> <article> <div class="publications"> <h2 class="bibliography">2025</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/sciadv_2025.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="sciadv_2025.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Shi2025" class="col-sm-8"> <div class="title">Single-nucleus multi-omics identifies shared and distinct pathways in Pick’s and Alzheimer’s disease</div> <div class="author"> Zechuan Shi, Sudeshna Das, <em>Samuel Morabito</em>, and <span class="more-authors" title="click to view 12 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '12 more authors' ? 'Jennifer Stocksdale, Emily Miyoshi, Shushrruth Sai Srinivasan, Nora Emerson, Arshi Shahin, Negin Rahimzadeh, Zhenkun Cao, Justine Silva, Andres A. Castaneda, Elizabeth Head, Leslie Thompson, Vivek Swarup' : '12 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">12 more authors</span> </div> <div class="periodical"> <em>Science Advances</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1126/sciadv.ads7973" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Shi2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1126/sciadv.ads7973"></span> <span class="__dimensions_badge_embed__" data-doi="10.1126/sciadv.ads7973" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The study of transcriptomic and epigenomic variations in neurodegenerative diseases, particularly tauopathies like Pick’s disease (PiD) and Alzheimer’s disease (AD), offers insights into their underlying regulatory mechanisms. Here, we identified critical regulatory changes driving disease progression, revealing potential therapeutic targets. Our comparative analyses uncovered disease-enriched noncoding regions and genome-wide transcription factor (TF) binding differences, linking them to target genes. Notably, we identified a distal human-gained enhancer (HGE) associated with E3 ubiquitin ligase (UBE3A), highlighting disease-specific regulatory alterations. Additionally, fine mapping of AD risk genes uncovered loci enriched in microglial enhancers and accessible in other cell types. Shared and distinct TF binding patterns were observed in neurons and glial cells across PiD and AD. We validated our findings using CRISPR to excise a predicted enhancer region in UBE3A and developed an interactive database, scROAD, to visualize predicted single-cell TF occupancy and regulatory networks.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/ad_july2025.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="ad_july2025.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="DSAD2025" class="col-sm-8"> <div class="title">Genetically determined Alzheimer’s disease research advances: The Down Syndrome &amp; Autosomal Dominant Alzheimer’s Disease 2024 Conference</div> <div class="author"> Neus Falgàs, Lucia Maure‐Blesa, Beau Ances, and <span class="more-authors" title="click to view 56 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '56 more authors' ? 'Lisi Flores‐Aguilar, Sigan Hartley, Jason Hassenstab, M Florencia Iulita, Matthew Janicki, Katherine Koenig, Patrick Lao, Johannes Levin, Eric McDade, Laurent Meijer, Michael S. Rafii, Heather M. Snyder, Raquel Sánchez‐Valle, Juan Fortea, DSAD‐ADAD conference group, Jose Arriola Infante, Mirea Balasa, Isabel Barroeta, Nicolas Barthelemy R, Alexandre Bejanin, Bessy Benejam, Beatriz Bosch, Angela Bradshaw, Maria Carmona‐Iragui, Ann Cohen, Aina Comas Albertí, Lajos Csincsik, A. Claudio Cuello, Laura del Hoyo Soriano, Janna Dijkstra, Natalie Edwards, Sandra Giménez, Fernando Gonzalez‐Ortiz, Brian Gordon, Sara Gutiérrez Fernández, Benjamin Handen, Charlotte Jacob, Erik Johnson, Charlotte Johansson, Albert Lladó, Alberto Lleó, Samuel Morabito, Alejandra. O. Morcillo‐Nieto, Laia Montoliu‐Gaya, Michael Okafor, Agnes Pérez‐Millan, Marie Claude Potier, John Ringman, Íñigo Rodríguez‐Baz, Eric Rubenstein, Natalie S. Ryan, André Strydom, Lidia Vaqué‐Alcázar, Lisa Vermunt, Laura Videla Toro, Shahid Zaman' : '56 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">56 more authors</span> </div> <div class="periodical"> <em>Alzheimer’s &amp; Dementia</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1002/alz.70309" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/DSAD2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1002/alz.70309"></span> <span class="__dimensions_badge_embed__" data-doi="10.1002/alz.70309" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The Down syndrome‐associated Alzheimer’s disease (DSAD) autosomal dominant Alzheimer’s disease (ADAD) 2024 Conference in Barcelona, convened under an Alzheimer’s Association International Society to Advance Alzheimer’s Research and Treatment (ISTAART) grant through the Down syndrome and Alzheimer’s disease (AD) Professional Interest Area (PIA), brought together global researchers to foster collaboration and knowledge exchange between the fields of DSAD and ADAD. This article provides a synthesis review of the conference proceedings, summarizing key discussions on biomarkers, natural history models, clinical trials, and ethical considerations in anti‐amyloid therapies. A total of 211 attendees from 16 countries joined the meeting. Global researchers presented on disease mechanisms, therapeutic developments, and patient care strategies. Discussions focused on challenges and opportunities unique to DSAD and ADAD. Experts emphasized the urgent need for tailored clinical trials for ADAD and DSAD and debated the safety and efficacy of anti‐amyloid treatments. Ethical considerations highlighted equitable access to therapies and the crucial role of patient and caregiver involvement. The conference highlighted the importance of inclusive research and collaboration across the genetic forms of AD. Biomarker research and natural history models developed in Down syndrome‐associated Alzheimer’s disease (DSAD) and autosomal dominant Alzheimer’s disease (ADAD) enable the prediction of disease progression not only for DSAD and ADAD, but also for sporadic Alzheimer’s disease (AD). ‐Collaboration and knowledge exchange among researchers across these genetic forms of AD will accelerate our understanding of the pathophysiology and advance preventive trials in DSAD and ADAD. ‐Tailored clinical trials for DSAD are urgently needed to address specific safety and efficacy concerns. ‐Inclusive research practices are crucial for advancing treatments and understanding of DSAD and ADAD.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/Dimas2025.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="Dimas2025.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Dimas2025" class="col-sm-8"> <div class="title">Dysregulation of transcription networks regulating oligodendrogenesis in age-related decline in CNS remyelination</div> <div class="author"> Penelope Dimas, <em>Samuel Morabito</em>, Khalil S Rawji, and <span class="more-authors" title="click to view 12 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '12 more authors' ? 'Qing Wang, Fynn N Krause, Juan Cubillos, Nejma Belaadi, Peggy Assinck, Zechuan Shi, Zhenkun Cao, Holger Heyn, Riki Kawaguchi, Daniel H Geschwind, Vivek Swarup, Robin JM Franklin' : '12 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">12 more authors</span> </div> <div class="periodical"> <em></em> 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2025.11.14.688494" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Dimas2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2025.11.14.688494"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2025.11.14.688494" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>In demyelinating diseases like multiple sclerosis (MS), efficient remyelination is critical for functional recovery. Remyelination efficiency declines with age, and is linked to progressive disability. The gene regulatory network underlying remyelination, and how it is altered with aging, remains unclear. Here we present a comparative single-nucleus RNA and ATAC sequencing analysis of remyelination in young and aged mice. We identified gene modules dynamically expressed throughout oligodendrocyte differentiation, revealing age-dependent changes in key processes related to myelination. Multi-omic analysis allowed us to map the regulatory network driving efficient remyelination within oligodendrocyte lineage cells in young mice. We highlight key transcription factors in the network dysregulated with age, and we describe similar dysregulations in MS lesions. Modifying the expression of these transcription factors in primary oligodendrocyte progenitor cells impacts differentiation. These findings provide a foundational understanding of this regenerative process in the context of aging and in chronic demyelinating diseases.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/Nieto2025.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="Nieto2025.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Nieto2025" class="col-sm-8"> <div class="title">Decoding the immune response in leptomeningeal disease through single-cell sequencing of cerebrospinal fluid</div> <div class="author"> Paula Nieto, Svenja Klinsing, Ginevra Caratù, and <span class="more-authors" title="click to view 19 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '19 more authors' ? 'Mareike Dettki, Domenica Marchese, Katharina J. Weber, Samuel Morabito, Patricia Lorden, Irene Ruano, Katharina Imkeller, M. Angels Velasco, Silvia Vidal, Juan L. Melero, Philipp Euskirchen, Marcus Czabanka, Karl H. Plate, Patrick N. Harter, Anna Pascual-Reguant, Joachim P. Steinbach, Holger Heyn, Pia S. Zeiner, Juan C. Nieto' : '19 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">19 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2025.01.27.634744" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Nieto2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2025.01.27.634744"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2025.01.27.634744" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Assessing anti-tumor immune responses and immune microenvironments in central nervous system (CNS) neoplasms, such as brain tumors and leptomeningeal disease (LMD), provides prognostic insights and predictive biomarkers. Liquid biopsy of the cerebrospinal fluid (CSF) represents a promising minimally-invasive approach, but its ability to reflect immune responses against tumors remains unclear. Here, we used single-cell sequencing of CSF cells and spatial transcriptomics of CNS lesions to compare and contrast LMD patients with CNS lymphoma (CNSL), glioblastoma (GB) and brain metastases (BrM), to neuroinflammatory CNS disorders. We identified disease-specific CSF environments, reflecting parenchymal tumor microenvironment features. CNSL showed robust T cell responses, while BrM and GB were dominated by both blood-derived and tissue-resident myeloid cells. Longitudinal CSF sampling unveiled mechanisms of disease progression and therapy resistance, highlighting the potential of CSF liquid biopsies for uncovering disease biology, discovering cellular biomarkers and developing personalized therapies for CNS neoplasms.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2024</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/natgen_2024.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="natgen_2024.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Miyoshi2024" class="col-sm-8"> <div class="title">Spatial and single-nucleus transcriptomic analysis of genetic and sporadic forms of Alzheimer’s disease</div> <div class="author"> Emily Miyoshi<sup>*</sup>, <em>Samuel Morabito<sup>*</sup></em>, Caden M Henningfield, and <span class="more-authors" title="click to view 24 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '24 more authors' ? 'Sudeshna Das, Negin Rahimzadeh, Sepideh Kiani Shabestari, Neethu Michael, Nora Emerson, Fairlie Reese, Zechuan Shi, Zhenkun Cao, Shushrruth Sai Srinivasan, Vanessa M Scarfone, Miguel A Arreola, Jackie Lu, Sierra Wright, Justine Silva, Kelsey Leavy, Ira T Lott, Eric Doran, William H Yong, Saba Shahin, Mari Perez-Rosendahl, Alzheimer’s Biomarkers Consortium–Down Syndrome (ABC–DS), Elizabeth Head, Kim N Green, Vivek Swarup' : '24 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">24 more authors</span> </div> <div class="periodical"> <em>Nature Genetics</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41588-024-01961-x" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Miyoshi_NatGen2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41588-024-01961-x"></span> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41588-024-01961-x" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The pathogenesis of Alzheimer’s disease (AD) depends on environmental and heritable factors, with its molecular etiology still unclear. Here we present a spatial transcriptomic (ST) and single-nucleus transcriptomic survey of late-onset sporadic AD and AD in Down syndrome (DSAD). Studying DSAD provides an opportunity to enhance our understanding of the AD transcriptome, potentially bridging the gap between genetic mouse models and sporadic AD. We identified transcriptomic changes that may underlie cortical layer-preferential pathology accumulation. Spatial co-expression network analyses revealed transient and regionally restricted disease processes, including a glial inflammatory program dysregulated in upper cortical layers and implicated in AD genetic risk and amyloid-associated processes. Cell–cell communication analysis further contextualized this gene program in dysregulated signaling networks. Finally, we generated ST data from an amyloid AD mouse model to identify cross-species amyloid-proximal transcriptomic changes with conformational context. Spatial and single-nucleus analyses in human postmortem Alzheimer’s disease (AD) brain tissues at early and late stages from individuals with and without Down syndrome, as well as in AD mouse models, show sex and species-specific phenotypic changes.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/cr_march2024.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cr_march2024.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Childs2024" class="col-sm-8"> <div class="title">Relapse to cocaine seeking is regulated by medial habenula NR4A2/NURR1 in mice</div> <div class="author"> Jessica E. Childs<sup>*</sup>, <em>Samuel Morabito<sup>*</sup></em>, Sudeshna Das, and <span class="more-authors" title="click to view 9 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '9 more authors' ? 'Caterina Santelli, Victoria Pham, Kelly Kusche, Vanessa Alizo Vera, Fairlie Reese, Rianne R. Campbell, Dina P. Matheos, Vivek Swarup, Marcelo A. Wood' : '9 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">9 more authors</span> </div> <div class="periodical"> <em>Cell Reports</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.celrep.2024.113956" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/childs2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.celrep.2024.113956"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.celrep.2024.113956" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Drugs of abuse can persistently change the reward circuit in ways that contribute to relapse behavior, partly via mechanisms that regulate chromatin structure and function. Nuclear orphan receptor subfamily4 groupA member2 (NR4A2, also known as NURR1) is an important effector of histone deacetylase 3 (HDAC3)-dependent mechanisms in persistent memory processes and is highly expressed in the medial habenula (MHb), a region that regulates nicotine-associated behaviors. Here, expressing the Nr4a2 dominant negative (Nurr2c) in the MHb blocks reinstatement of cocaine seeking in mice. We use single-nucleus transcriptomics to characterize the molecular cascade following Nr4a2 manipulation, revealing changes in transcriptional networks related to addiction, neuroplasticity, and GABAergic and glutamatergic signaling. The network controlled by NR4A2 is characterized using a transcription factor regulatory network inference algorithm. These results identify the MHb as a pivotal regulator of relapse behavior and demonstrate the importance of NR4A2 as a key mechanism driving the MHb component of relapse.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/ad_2024.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="ad_2024.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Agudo2024" class="col-sm-8"> <div class="title">BIN1K358R suppresses glial response to plaques in mouse model of Alzheimer’s disease</div> <div class="author"> Laura Fernandez Garcia‐Agudo, Zechuan Shi, Ian F. Smith, and <span class="more-authors" title="click to view 24 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '24 more authors' ? 'Enikö A. Kramár, Katelynn Tran, Shimako Kawauchi, Shuling Wang, Sherilyn Collins, Amber Walker, Kai‐Xuan Shi, Jonathan Neumann, Heidi Yahan Liang, Celia Da Cunha, Giedre Milinkeviciute, Samuel Morabito, Emily Miyoshi, Narges Rezaie, Angela Gomez‐Arboledas, Adrian Mendoza Arvilla, Daryan Iman Ghaemi, Andrea J. Tenner, Frank M. LaFerla, Marcelo A. Wood, Ali Mortazavi, Vivek Swarup, Grant R. MacGregor, Kim N. Green' : '24 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">24 more authors</span> </div> <div class="periodical"> <em>Alzheimer’s &amp; Dementia</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1002/alz.13767" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Agudo2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1002/alz.13767"></span> <span class="__dimensions_badge_embed__" data-doi="10.1002/alz.13767" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The BIN1 coding variant rs138047593 (K358R) is linked to Late‐Onset Alzheimer’s Disease (LOAD) via targeted exome sequencing. To elucidate the functional consequences of this rare coding variant on brain amyloidosis and neuroinflammation, we generated BIN1K358R knock‐in mice using CRISPR/Cas9 technology. These mice were subsequently bred with 5xFAD transgenic mice, which serve as a model for Alzheimer’s pathology. The presence of the BIN1K358R variant leads to increased cerebral amyloid deposition, with a dampened response of astrocytes and oligodendrocytes, but not microglia, at both the cellular and transcriptional levels. This correlates with decreased neurofilament light chain in both plasma and brain tissue. Synaptic densities are significantly increased in both wild‐type and 5xFAD backgrounds homozygous for the BIN1K358R variant. The BIN1 K358R variant modulates amyloid pathology in 5xFAD mice, attenuates the astrocytic and oligodendrocytic responses to amyloid plaques, decreases damage markers, and elevates synaptic densities. BIN1 rs138047593 (K358R) coding variant is associated with increased risk of LOAD. BIN1 K358R variant increases amyloid plaque load in 12‐month‐old 5xFAD mice. BIN1 K358R variant dampens astrocytic and oligodendrocytic response to plaques. BIN1 K358R variant decreases neuronal damage in 5xFAD mice. BIN1 K358R upregulates synaptic densities and modulates synaptic transmission.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/Rebboah2024.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="Rebboah2024.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Rebboah2024" class="col-sm-8"> <div class="title">The ENCODE mouse postnatal developmental time course identifies regulatory programs of cell types and cell states</div> <div class="author"> Elisabeth Rebboah, Narges Rezaie, Brian A Williams, and <span class="more-authors" title="click to view 14 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '14 more authors' ? 'Annika K Weimer, Minyi Shi, Xinqiong Yang, Heidi Yahan Liang, Louise Dionne, Fairlie Reese, Diane Trout, Jennifer Jou, Ingrid Youngworth, Laura G Reinholdt, Samuel Morabito, Michael Snyder, Barbara Wold, Ali Mortazavi' : '14 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">14 more authors</span> </div> <div class="periodical"> <em></em> 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2024.06.12.598567" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/Rebboah2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2024.06.12.598567"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2024.06.12.598567" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Postnatal genomic regulation significantly influences tissue and organ maturation but is under-studied relative to existing genomic catalogs of adult tissues or prenatal development in mouse. The ENCODE4 consortium generated the first comprehensive single-nucleus resource of postnatal regulatory events across a diverse set of mouse tissues. The collection spans seven postnatal time points, mirroring human development from childhood to adulthood, and encompasses five core tissues. We identified 30 cell types, further subdivided into 69 subtypes and cell states across adrenal gland, left cerebral cortex, hippocampus, heart, and gastrocnemius muscle. Our annotations cover both known and novel cell differentiation dynamics ranging from early hippocampal neurogenesis to a new sex-specific adrenal gland population during puberty. We used an ensemble Latent Dirichlet Allocation strategy with a curated vocabulary of 2,701 regulatory genes to identify regulatory "topics," each of which is a gene vector, linked to cell type differentiation, subtype specialization, and transitions between cell states. We find recurrent regulatory topics in tissue-resident macrophages, neural cell types, endothelial cells across multiple tissues, and cycling cells of the adrenal gland and heart. Cell-type-specific topics are enriched in transcription factors and microRNA host genes, while chromatin regulators dominate mitosis topics. Corresponding chromatin accessibility data reveal dynamic and sex-specific regulatory elements, with enriched motifs matching transcription factors in regulatory topics. Together, these analyses identify both tissue-specific and common regulatory programs in postnatal development across multiple tissues through the lens of the factors regulating transcription.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/cellrepmeth_2024.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cellrepmeth_2024.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito2023" class="col-sm-8"> <div class="title">hdWGCNA identifies co-expression networks in high-dimensional transcriptomics data</div> <div class="author"> <em>Samuel Morabito</em>, Fairlie Reese, Negin Rahimzadeh, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Emily Miyoshi, Vivek Swarup' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Cell Reports Methods</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.crmeth.2023.100498" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/morabito_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.crmeth.2023.100498"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.crmeth.2023.100498" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Biological systems are immensely complex, organized into a multi-scale hierarchy of functional units based on tightly regulated interactions between distinct molecules, cells, organs, and organisms. While experimental methods enable transcriptome-wide measurements across millions of cells, popular bioinformatic tools do not support systems-level analysis. Here we present hdWGCNA, a comprehensive framework for analyzing co-expression networks in high-dimensional transcriptomics data such as single-cell and spatial RNA sequencing (RNA-seq). hdWGCNA provides functions for network inference, gene module identification, gene enrichment analysis, statistical tests, and data visualization. Beyond conventional single-cell RNA-seq, hdWGCNA is capable of performing isoform-level network analysis using long-read single-cell data. We showcase hdWGCNA using data from autism spectrum disorder and Alzheimer’s disease brain samples, identifying disease-relevant co-expression network modules. hdWGCNA is directly compatible with Seurat, a widely used R package for single-cell and spatial transcriptomics analysis, and we demonstrate the scalability of hdWGCNA by analyzing a dataset containing nearly 1 million cells</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/Reese2023.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="Reese2023.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Reese2023" class="col-sm-8"> <div class="title">The ENCODE4 long-read RNA-seq collection reveals distinct classes of transcript structure diversity</div> <div class="author"> Fairlie Reese, Brian Williams, Gabriela Balderrama-Gutierrez, and <span class="more-authors" title="click to view 42 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '42 more authors' ? 'Dana Wyman, Muhammed Hasan Çelik, Elisabeth Rebboah, Narges Rezaie, Diane Trout, Milad Razavi-Mohseni, Yunzhe Jiang, Beatrice Borsari, Samuel Morabito, Heidi Yahan Liang, Cassandra J McGill, Sorena Rahmanian, Jasmine Sakr, Shan Jiang, Weihua Zeng, Klebea Carvalho, Annika K Weimer, Louise A Dionne, Ariel McShane, Karan Bedi, Shaimae I Elhajjajy, Sean Upchurch, Jennifer Jou, Ingrid Youngworth, Idan Gabdank, Paul Sud, Otto Jolanki, J Seth Strattan, Meenakshi S Kagda, Michael P Snyder, Ben C Hitz, Jill E Moore, Zhiping Weng, David Bennett, Laura Reinholdt, Mats Ljungman, Michael A Beer, Mark B Gerstein, Lior Pachter, Roderic Guigó, Barbara J Wold, Ali Mortazavi' : '42 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">42 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2023.05.15.540865" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/reese2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1101/2023.05.15.540865"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.05.15.540865" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The majority of mammalian genes encode multiple transcript isoforms that result from differential promoter use, changes in exonic splicing, and alternative 3′ end choice. Detecting and quantifying transcript isoforms across tissues, cell types, and species has been extremely challenging because transcripts are much longer than the short reads normally used for RNA-seq. By contrast, long-read RNA-seq (LR-RNA-seq) gives the complete structure of most transcripts. We sequenced 264 LR-RNA-seq PacBio libraries totaling over 1 billion circular consensus reads (CCS) for 81 unique human and mouse samples. We detect at least one full-length transcript from 87.7% of annotated human protein coding genes and a total of \textbackslashtextasciitilde200,000 full-length transcripts, \textbackslashtextasciitilde40% of which have novel exon junction chains. To capture and compute on the three sources of transcript structure diversity, we introduce a gene and transcript annotation framework that uses triplets representing the transcript start site, exon junction chain, and transcript end site of each transcript. Using triplets in a simplex representation demonstrates how promoter selection, splice pattern, and 3′ processing are deployed across human tissues, with nearly half of multitranscript protein coding genes showing a clear bias toward one of the three diversity mechanisms. Evaluated across samples, the predominantly expressed transcript changes for 74% of protein coding genes. In evolution, the human and mouse transcriptomes are globally similar in types of transcript structure diversity, yet among individual orthologous gene pairs, more than half (57.8%) show substantial differences in mechanism of diversification in matching tissues. This initial large-scale survey of human and mouse long-read transcriptomes provides a foundation for further analyses of alternative transcript usage, and is complemented by short-read and microRNA data on the same samples and by epigenome data elsewhere in the ENCODE4 collection.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/cellmetab_2023.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cellmetab_2023.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Costanzo.2023.Cell Metabolism" class="col-sm-8"> <div class="title">The Type 2 Diabetes Knowledge Portal: An open access genetic resource dedicated to type 2 diabetes and related traits</div> <div class="author"> Maria C. Costanzo, Marcin von Grotthuss, Jeffrey Massung, and <span class="more-authors" title="click to view 189 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '189 more authors' ? 'Dongkeun Jang, Lizz Caulkins, Ryan Koesterer, Clint Gilbert, Ryan P. Welch, Parul Kudtarkar, Quy Hoang, Andrew P. Boughton, Preeti Singh, Ying Sun, Marc Duby, Annie Moriondo, Trang Nguyen, Patrick Smadbeck, Benjamin R. Alexander, MacKenzie Brandes, Mary Carmichael, Peter Dornbos, Todd Green, Kenneth C. Huellas-Bruskiewicz, Yue Ji, Alexandria Kluge, Aoife C. McMahon, Josep M. Mercader, Oliver Ruebenacker, Sebanti Sengupta, Dylan Spalding, Daniel Taliun, AMP-T2D Consortium, Gonçalo Abecasis, Beena Akolkar, Benjamin R. Alexander, Nicholette D. Allred, David Altshuler, Jennifer E. Below, Richard Bergman, Joline W.J. Beulens, John Blangero, Michael Boehnke, Krister Bokvist, Erwin Bottinger, Andrew P. Boughton, Donald Bowden, M. Julia Brosnan, Christopher Brown, Kenneth Bruskiewicz, Noël P. Burtt, Mary Carmichael, Lizz Caulkins, Inês Cebola, John Chambers, Yii-Der Ida Chen, Andriy Cherkas, Audrey Y. Chu, Christopher Clark, Melina Claussnitzer, Maria C. Costanzo, Nancy J. Cox, Marcel den Hoed, Duc Dong, Marc Duby, Ravindranath Duggirala, Josée Dupuis, Petra J.M. Elders, Jesse M. Engreitz, Eric Fauman, Jorge Ferrer, Jason Flannick, Paul Flicek, Matthew Flickinger, Jose C. Florez, Caroline S. Fox, Timothy M. Frayling, Kelly A. Frazer, Kyle J. Gaulton, Clint Gilbert, Anna L. Gloyn, Todd Green, Craig L. Hanis, Robert Hanson, Andrew T. Hattersley, Quy Hoang, Hae Kyung Im, Sidra Iqbal, Suzanne B.R. Jacobs, Dong-Keun Jang, Tad Jordan, Tania Kamphaus, Fredrik Karpe, Thomas M. Keane, Seung K. Kim, Alexandria Kluge, Ryan Koesterer, Parul Kudtarkar, Kasper Lage, Leslie A. Lange, Mitchell Lazar, Donna Lehman, Ching-Ti Liu, Ruth J.F. Loos, Ronald Ching-wan Ma, Patrick MacDonald, Jeffrey Massung, Matthew T. Maurano, Mark I. McCarthy, Gil McVean, James B. Meigs, Josep M. Mercader, Melissa R. Miller, Braxton Mitchell, Karen L. Mohlke, Samuel Morabito, Claire Morgan, Shannon Mullican, Sharvari Narendra, Maggie C.Y. Ng, Lynette Nguyen, Colin N.A. Palmer, Stephen C.J. Parker, Antonio Parrado, Afshin Parsa, Aaron C. Pawlyk, Ewan R. Pearson, Andrew Plump, Michael Province, Thomas Quertermous, Susan Redline, Dermot F. Reilly, Bing Ren, Stephen S. Rich, J. Brent Richards, Jerome I. Rotter, Oliver Ruebenacker, Hartmut Ruetten, Rany M. Salem, Maike Sander, Michael Sanders, Dharambir Sanghera, Laura J. Scott, Sebanti Sengupta, David Siedzik, Xueling Sim, Preeti Singh, Robert Sladek, Kerrin Small, Philip Smith, Peter Stein, Dylan Spalding, Heather M. Stringham, Ying Sun, Katalin Susztak, Leen M. ’t Hart, Daniel Taliun, Kent Taylor, Melissa K. Thomas, Jennifer A. Todd, Miriam S. Udler, Benjamin Voight, Marcin von Grotthuss, Andre Wan, Ryan P. Welch, David Wholley, Kaan Yuksel, Norann A. Zaghloul, Philip Smith, Melissa K. Thomas, Beena Akolkar, M. Julia Brosnan, Andriy Cherkas, Audrey Y. Chu, Eric B. Fauman, Caroline S. Fox, Tania Nayak Kamphaus, Melissa R. Miller, Lynette Nguyen, Afshin Parsa, Dermot F. Reilly, Hartmut Ruetten, David Wholley, Norann A. Zaghloul, Gonçalo R. Abecasis, David Altshuler, Thomas M. Keane, Mark I. McCarthy, Kyle J. Gaulton, Jose C. Florez, Michael Boehnke, Noël P. Burtt, Jason Flannick' : '189 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">189 more authors</span> </div> <div class="periodical"> <em>Cell Metabolism</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.cmet.2023.03.001" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/costanzo_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.cmet.2023.03.001"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.cmet.2023.03.001" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Associations between human genetic variation and clinical phenotypes have become a foundation of biomedical research. Most repositories of these data seek to be disease-agnostic and therefore lack disease-focused views. The Type 2 Diabetes Knowledge Portal (T2DKP) is a public resource of genetic datasets and genomic annotations dedicated to type 2 diabetes (T2D) and related traits. Here, we seek to make the T2DKP more accessible to prospective users and more useful to existing users. First, we evaluate the T2DKP’s comprehensiveness by comparing its datasets with those of other repositories. Second, we describe how researchers unfamiliar with human genetic data can begin using and correctly interpreting them via the T2DKP. Third, we describe how existing users can extend their current workflows to use the full suite of tools offered by the T2DKP. We finally discuss the lessons offered by the T2DKP toward the goal of democratizing access to complex disease genetic results.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/neuron_2022.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="neuron_2022.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="OteroGarcia2022" class="col-sm-8"> <div class="title">Molecular signatures underlying neurofibrillary tangle susceptibility in Alzheimer’s disease</div> <div class="author"> Marcos Otero-Garcia, Sameehan U. Mahajani, Debia Wakhloo, and <span class="more-authors" title="click to view 14 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '14 more authors' ? 'Weijing Tang, Yue-Qiang Xue, Samuel Morabito, Jie Pan, Jane Oberhauser, Angela E. Madira, Tamara Shakouri, Yongning Deng, Thomas Allison, Zihuai He, William E. Lowry, Riki Kawaguchi, Vivek Swarup, Inma Cobos' : '14 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">14 more authors</span> </div> <div class="periodical"> <em>Neuron</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.neuron.2022.06.021" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/otero_garcia_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.neuron.2022.06.021"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.neuron.2022.06.021" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Tau aggregation in neurofibrillary tangles (NFTs) is closely associated with neurodegeneration and cognitive decline in Alzheimer’s disease (AD). However, the molecular signatures that distinguish between aggregation-prone and aggregation-resistant cell states are unknown. We developed methods for the high-throughput isolation and transcriptome profiling of single somas with NFTs from the human AD brain, quantified the susceptibility of 20 neocortical subtypes for NFT formation and death, and identified both shared and cell-type-specific signatures. NFT-bearing neurons shared a marked upregulation of synaptic transmission-related genes, including a core set of 63 genes enriched for synaptic vesicle cycling. Oxidative phosphorylation and mitochondrial dysfunction were highly cell-type dependent. Apoptosis was only modestly enriched, and the susceptibilities of NFT-bearing and NFT-free neurons for death were highly similar. Our analysis suggests that NFTs represent cell-type-specific responses to stress and synaptic dysfunction. We provide a resource for biomarker discovery and the investigation of tau-dependent and tau-independent mechanisms of neurodegeneration.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/cr_2022.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cr_2022.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Shabestari2022" class="col-sm-8"> <div class="title">Absence of microglia promotes diverse pathologies and early lethality in Alzheimer’s disease mice</div> <div class="author"> Sepideh Kiani Shabestari, <em>Samuel Morabito</em>, Emma Pascal Danhash, and <span class="more-authors" title="click to view 18 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '18 more authors' ? 'Amanda McQuade, Jessica Ramirez Sanchez, Emily Miyoshi, Jean Paul Chadarevian, Christel Claes, Morgan Alexandra Coburn, Jonathan Hasselmann, Jorge Hidalgo, Kayla Nhi Tran, Alessandra C. Martini, Winston Chang Rothermich, Jesse Pascual, Elizabeth Head, David A. Hume, Clare Pridans, Hayk Davtyan, Vivek Swarup, Mathew Blurton-Jones' : '18 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">18 more authors</span> </div> <div class="periodical"> <em>Cell Reports</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.celrep.2022.110961" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/shabestari_cr2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.celrep.2022.110961"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.celrep.2022.110961" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Microglia are strongly implicated in the development and progression of Alzheimer’s disease (AD), yet their impact on pathology and lifespan remains unclear. Here we utilize a CSF1R hypomorphic mouse to generate a model of AD that genetically lacks microglia. The resulting microglial-deficient mice exhibit a profound shift from parenchymal amyloid plaques to cerebral amyloid angiopathy (CAA), which is accompanied by numerous transcriptional changes, greatly increased brain calcification and hemorrhages, and premature lethality. Remarkably, a single injection of wild-type microglia into adult mice repopulates the microglial niche and prevents each of these pathological changes. Taken together, these results indicate the protective functions of microglia in reducing CAA, blood-brain barrier dysfunction, and brain calcification. To further understand the clinical implications of these findings, human AD tissue and iPSC-microglia were examined, providing evidence that microglia phagocytose calcium crystals, and this process is impaired by loss of the AD risk gene, TREM2.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/starprotoc_2022.gif" class="preview z-depth-1 rounded" width="100%" height="auto" alt="starprotoc_2022.gif" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="10.1016/j.xpro.2022.101491" class="col-sm-8"> <div class="title">Protocol for single-nucleus ATAC sequencing and bioinformatic analysis in frozen human brain tissue</div> <div class="author"> Zechuan Shi<sup>*</sup>, Sudeshna Das<sup>*</sup>, <em>Samuel Morabito</em>, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Emily Miyoshi, Vivek Swarup' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>STAR Protocols</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.xpro.2022.101491" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/shi_star_protocol_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.xpro.2022.101491" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Single-nucleus ATAC sequencing (snATAC-seq) employs a hyperactive Tn5 transposase to gain precise information about the cis-regulatory elements in specific cell types. However, the standard protocol of snATAC-seq is not optimized for all tissues, including the brain. Here, we present a modified protocol for single-nuclei isolation from postmortem frozen human brain tissue, followed by snATAC-seq library preparation and sequencing. We also describe an integrated bioinformatics analysis pipeline using an R package (ArchRtoSignac) to robustly analyze snATAC-seq data. For complete details on the use and execution of this protocol, please refer to Morabito et al. (2021).</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/SciRep_2022.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="SciRep_2022.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Barahona2022" class="col-sm-8"> <div class="title">Cortical diurnal rhythms remain intact with microglial depletion</div> <div class="author"> Rocio A. Barahona, <em>Samuel Morabito</em>, Vivek Swarup, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Kim N. Green' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Scientific Reports</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41598-021-04079-w" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/barahona_SciRep_2022.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41598-021-04079-w" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Microglia are subject to change in tandem with the endogenously generated biological oscillations known as our circadian rhythm. Studies have shown microglia harbor an intrinsic molecular clock which regulates diurnal changes in morphology and influences inflammatory responses. In the adult brain, microglia play an important role in the regulation of condensed extracellular matrix structures called perineuronal nets (PNNs), and it has been suggested that PNNs are also regulated in a circadian and diurnal manner. We sought to determine whether microglia mediate the diurnal regulation of PNNs via CSF1R inhibitor dependent microglial depletion in C57BL/6J mice, and how the absence of microglia might affect cortical diurnal gene expression rhythms. While we observe diurnal differences in microglial morphology, where microglia are most ramified at the onset of the dark phase, we do not find diurnal differences in PNN intensity. However, PNN intensity increases across many brain regions in the absence of microglia, supporting a role for microglia in the regulation of PNNs. Here, we also show that cortical diurnal gene expression rhythms are intact, with no cycling gene changes without microglia. These findings demonstrate a role for microglia in the maintenance of PNNs, but not in the maintenance of diurnal rhythms.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/natgen_july2021.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="natgen_july2021.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito2021" class="col-sm-8"> <div class="title">Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito<sup>*</sup></em>, Emily Miyoshi<sup>*</sup>, Neethu Michael, and <span class="more-authors" title="click to view 7 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '7 more authors' ? 'Saba Shahin, Alessandra Cadete Martini, Elizabeth Head, Justine Silva, Kelsey Leavy, Mari Perez-Rosendahl, Vivek Swarup' : '7 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">7 more authors</span> </div> <div class="periodical"> <em>Nature Genetics</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41588-021-00894-z" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/morabito_NatGen_2021.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41588-021-00894-z"></span> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41588-021-00894-z" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The gene-regulatory landscape of the brain is highly dynamic in health and disease, coordinating a menagerie of biological processes across distinct cell types. Here, we present a multi-omic single-nucleus study of 191,890 nuclei in late-stage Alzheimer’s disease (AD), accessible through our web portal, profiling chromatin accessibility and gene expression in the same biological samples and uncovering vast cellular heterogeneity. We identified cell-type-specific, disease-associated candidate cis-regulatory elements and their candidate target genes, including an oligodendrocyte-associated regulatory module containing links to APOE and CLU. We describe cis-regulatory relationships in specific cell types at a subset of AD risk loci defined by genome-wide association studies, demonstrating the utility of this multi-omic single-nucleus approach. Trajectory analysis of glial populations identified disease-relevant transcription factors, such as SREBF1, and their regulatory targets. Finally, we introduce single-nucleus consensus weighted gene coexpression analysis, a coexpression network analysis strategy robust to sparse single-cell data, and perform a systems-level analysis of the AD transcriptome.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/nbd_2022.gif" class="preview z-depth-1 rounded" width="100%" height="auto" alt="nbd_2022.gif" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Miyoshi.NBD.2021" class="col-sm-8"> <div class="title">Systems biology approaches to unravel the molecular and genetic architecture of Alzheimer’s disease and related tauopathies</div> <div class="author"> Emily Miyoshi<sup>*</sup>, <em>Samuel Morabito<sup>*</sup></em>, and Vivek Swarup </div> <div class="periodical"> <em>Neurobiology of Disease</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.nbd.2021.105530" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/miyoshi_nbd_2021.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.nbd.2021.105530" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Over the years, genetic studies have identified multiple genetic risk variants associated with neurodegenerative disorders and helped reveal new biological pathways and genes of interest. However, genetic risk variants commonly reside in non-coding regions and may regulate distant genes rather than the nearest gene, as well as a gene’s interaction partners in biological networks. Systems biology and functional genomics approaches provide the framework to unravel the functional significance of genetic risk variants in disease. In this review, we summarize the genetic and transcriptomic studies of Alzheimer’s disease and related tauopathies and focus on the advantages of performing systems-level analyses to interrogate the biological pathways underlying neurodegeneration. Finally, we highlight new avenues of multi-omics analysis with single-cell approaches, which provides unparalleled opportunities to systematically explore cellular heterogeneity, and present an example of how to integrate publicly available single-cell datasets. Systems-level analysis has illuminated the function of many disease risk genes, but much work remains to study tauopathies and to understand spatiotemporal gene expression changes of specific cell types.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/hmg_2020.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="hmg_2020.jpeg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Morabito.HMG.2020" class="col-sm-8"> <div class="title">Integrative genomics approach identifies conserved transcriptomic networks in Alzheimer’s disease</div> <div class="author"> <em>Samuel Morabito</em>, Emily Miyoshi, Neethu Michael, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Vivek Swarup' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Human Molecular Genetics</em>, Aug 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1093/hmg/ddaa182" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/morabito-HMG-2020.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="/assets/pdf/morabito-HMG-2020-supp.pdf" class="btn btn-sm z-depth-0" role="button">Supp</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1093/hmg/ddaa182" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Alzheimer’s disease (AD) is a devastating neurological disorder characterized by changes in cell-type proportions and consequently marked alterations of the transcriptome. Here we use a data-driven systems biology meta-analytical approach across three human ad cohorts, encompassing six cortical brain regions, and integrate with multi-scale datasets comprising of DNA methylation, histone acetylation, transcriptome- and genome-wide association studies, and quantitative trait loci to further characterize the genetic architecture of ad. We perform co-expression network analysis across more than twelve hundred human brain samples, identifying robust ad-associated dysregulation of the transcriptome, unaltered in normal human aging. We assess the cell-type specificity of ad gene co-expression changes and estimate cell-type proportion changes in human ad by integrating co-expression modules with single-cell transcriptome data generated from 27 321 nuclei from human postmortem prefrontal cortical tissue. We also show that genetic variants of ad are enriched in a microglial ad-associated module and identify key transcription factors regulating co-expressed modules. Additionally, we validate our results in multiple published human ad gene expression datasets, which can be easily accessed using our online resource (https://swaruplab.bio.uci.edu/consensusAD).</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/Chen2019.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="Chen2019.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Chen2019" class="col-sm-8"> <div class="title">Single-cell landscape in mammary epithelium reveals bipotent-like cells associated with breast cancer risk and outcome</div> <div class="author"> Weiyan Chen, Samuel J. Morabito, Kai Kessenbrock, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Tariq Enver, Kerstin B. Meyer, Andrew E. Teschendorff' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>Communications Biology</em>, Dec 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s42003-019-0554-8" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a href="/assets/pdf/commbio_2019.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="__dimensions_badge_embed__" data-doi="10.1038/s42003-019-0554-8" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Adult stem-cells may serve as the cell-of-origin for cancer, yet their unbiased identification in single cell RNA sequencing data is challenging due to the high dropout rate. In the case of breast, the existence of a bipotent stem-like state is also controversial. Here we apply a marker-free algorithm to scRNA-Seq data from the human mammary epithelium, revealing a high-potency cell-state enriched for an independent mammary stem-cell expression module. We validate this stem-like state in independent scRNA-Seq data. Our algorithm further predicts that the stem-like state is bipotent, a prediction we are able to validate using FACS sorted bulk expression data. The bipotent stem-like state correlates with clinical outcome in basal breast cancer and is characterized by overexpression of YBX1 and ENO1, two modulators of basal breast cancer risk. This study illustrates the power of a marker-free computational framework to identify a novel bipotent stem-like state in the mammary epithelium.</p> </div> </div> </div> </li></ol> </div> </article> </div> </div> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?c15de51d4bb57887caa2c21988d97279"></script> <script defer src="/assets/js/copy_code.js?c8a01c11a92744d44b093fc3bda915df" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?a5bb4e6a542c546dd929b24b8b236dfd"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>